From 5820c3731ed51c9473315a6d370eeb2bf199cca5 Mon Sep 17 00:00:00 2001
From: Alistair Francis <alistair.francis@wdc.com>
Date: Mon, 13 Jul 2020 09:08:25 -0700
Subject: [PATCH 062/267] RISC-V: Add 32-bit ABI lists

Use the update-abi Make target to generate the abilist for RV32.

Reviewed-by: Maciej W. Rozycki <macro@wdc.com>
---
 .../unix/sysv/linux/riscv/rv32/c++-types.data |   67 +
 sysdeps/unix/sysv/linux/riscv/rv32/ld.abilist |    5 +
 .../linux/riscv/rv32/libBrokenLocale.abilist  |    1 +
 .../unix/sysv/linux/riscv/rv32/libanl.abilist |    4 +
 .../unix/sysv/linux/riscv/rv32/libc.abilist   | 1935 +++++++++++++++++
 .../sysv/linux/riscv/rv32/libcrypt.abilist    |    2 +
 .../unix/sysv/linux/riscv/rv32/libdl.abilist  |    9 +
 .../unix/sysv/linux/riscv/rv32/libm.abilist   |  940 ++++++++
 .../sysv/linux/riscv/rv32/libpthread.abilist  |  213 ++
 .../sysv/linux/riscv/rv32/libresolv.abilist   |   79 +
 .../unix/sysv/linux/riscv/rv32/librt.abilist  |   35 +
 .../linux/riscv/rv32/libthread_db.abilist     |   40 +
 .../sysv/linux/riscv/rv32/libutil.abilist     |    6 +
 13 files changed, 3336 insertions(+)
 create mode 100644 sysdeps/unix/sysv/linux/riscv/rv32/c++-types.data
 create mode 100644 sysdeps/unix/sysv/linux/riscv/rv32/ld.abilist
 create mode 100644 sysdeps/unix/sysv/linux/riscv/rv32/libBrokenLocale.abilist
 create mode 100644 sysdeps/unix/sysv/linux/riscv/rv32/libanl.abilist
 create mode 100644 sysdeps/unix/sysv/linux/riscv/rv32/libc.abilist
 create mode 100644 sysdeps/unix/sysv/linux/riscv/rv32/libcrypt.abilist
 create mode 100644 sysdeps/unix/sysv/linux/riscv/rv32/libdl.abilist
 create mode 100644 sysdeps/unix/sysv/linux/riscv/rv32/libm.abilist
 create mode 100644 sysdeps/unix/sysv/linux/riscv/rv32/libpthread.abilist
 create mode 100644 sysdeps/unix/sysv/linux/riscv/rv32/libresolv.abilist
 create mode 100644 sysdeps/unix/sysv/linux/riscv/rv32/librt.abilist
 create mode 100644 sysdeps/unix/sysv/linux/riscv/rv32/libthread_db.abilist
 create mode 100644 sysdeps/unix/sysv/linux/riscv/rv32/libutil.abilist

diff --git a/sysdeps/unix/sysv/linux/riscv/rv32/c++-types.data b/sysdeps/unix/sysv/linux/riscv/rv32/c++-types.data
new file mode 100644
index 0000000000..0fb0143ae7
--- /dev/null
+++ b/sysdeps/unix/sysv/linux/riscv/rv32/c++-types.data
@@ -0,0 +1,67 @@
+blkcnt64_t:x
+blkcnt_t:x
+blksize_t:i
+caddr_t:Pc
+clockid_t:i
+clock_t:l
+daddr_t:i
+dev_t:y
+fd_mask:l
+fsblkcnt64_t:y
+fsblkcnt_t:y
+fsfilcnt64_t:y
+fsfilcnt_t:y
+fsid_t:8__fsid_t
+gid_t:j
+id_t:j
+ino64_t:y
+ino_t:y
+int16_t:s
+int32_t:i
+int64_t:x
+int8_t:a
+intptr_t:i
+key_t:i
+loff_t:x
+mode_t:j
+nlink_t:j
+off64_t:x
+off_t:x
+pid_t:i
+pthread_attr_t:14pthread_attr_t
+pthread_barrier_t:17pthread_barrier_t
+pthread_barrierattr_t:21pthread_barrierattr_t
+pthread_cond_t:14pthread_cond_t
+pthread_condattr_t:18pthread_condattr_t
+pthread_key_t:j
+pthread_mutex_t:15pthread_mutex_t
+pthread_mutexattr_t:19pthread_mutexattr_t
+pthread_once_t:i
+pthread_rwlock_t:16pthread_rwlock_t
+pthread_rwlockattr_t:20pthread_rwlockattr_t
+pthread_spinlock_t:i
+pthread_t:m
+quad_t:x
+register_t:i
+rlim64_t:y
+rlim_t:y
+sigset_t:10__sigset_t
+size_t:j
+socklen_t:j
+ssize_t:i
+suseconds_t:x
+time_t:x
+u_char:h
+uid_t:j
+uint:j
+u_int:j
+u_int16_t:t
+u_int32_t:j
+u_int64_t:y
+u_int8_t:h
+ulong:m
+u_long:m
+u_quad_t:y
+useconds_t:j
+ushort:t
+u_short:t
diff --git a/sysdeps/unix/sysv/linux/riscv/rv32/ld.abilist b/sysdeps/unix/sysv/linux/riscv/rv32/ld.abilist
new file mode 100644
index 0000000000..94ca64c43d
--- /dev/null
+++ b/sysdeps/unix/sysv/linux/riscv/rv32/ld.abilist
@@ -0,0 +1,5 @@
+GLIBC_2.33 __libc_stack_end D 0x4
+GLIBC_2.33 __stack_chk_guard D 0x4
+GLIBC_2.33 __tls_get_addr F
+GLIBC_2.33 _dl_mcount F
+GLIBC_2.33 _r_debug D 0x14
diff --git a/sysdeps/unix/sysv/linux/riscv/rv32/libBrokenLocale.abilist b/sysdeps/unix/sysv/linux/riscv/rv32/libBrokenLocale.abilist
new file mode 100644
index 0000000000..cc8a5095f4
--- /dev/null
+++ b/sysdeps/unix/sysv/linux/riscv/rv32/libBrokenLocale.abilist
@@ -0,0 +1 @@
+GLIBC_2.33 __ctype_get_mb_cur_max F
diff --git a/sysdeps/unix/sysv/linux/riscv/rv32/libanl.abilist b/sysdeps/unix/sysv/linux/riscv/rv32/libanl.abilist
new file mode 100644
index 0000000000..33f23fd896
--- /dev/null
+++ b/sysdeps/unix/sysv/linux/riscv/rv32/libanl.abilist
@@ -0,0 +1,4 @@
+GLIBC_2.33 gai_cancel F
+GLIBC_2.33 gai_error F
+GLIBC_2.33 gai_suspend F
+GLIBC_2.33 getaddrinfo_a F
diff --git a/sysdeps/unix/sysv/linux/riscv/rv32/libc.abilist b/sysdeps/unix/sysv/linux/riscv/rv32/libc.abilist
new file mode 100644
index 0000000000..3bea073169
--- /dev/null
+++ b/sysdeps/unix/sysv/linux/riscv/rv32/libc.abilist
@@ -0,0 +1,1935 @@
+GLIBC_2.33 _Exit F
+GLIBC_2.33 _IO_2_1_stderr_ D 0xa0
+GLIBC_2.33 _IO_2_1_stdin_ D 0xa0
+GLIBC_2.33 _IO_2_1_stdout_ D 0xa0
+GLIBC_2.33 _IO_adjust_column F
+GLIBC_2.33 _IO_adjust_wcolumn F
+GLIBC_2.33 _IO_default_doallocate F
+GLIBC_2.33 _IO_default_finish F
+GLIBC_2.33 _IO_default_pbackfail F
+GLIBC_2.33 _IO_default_uflow F
+GLIBC_2.33 _IO_default_xsgetn F
+GLIBC_2.33 _IO_default_xsputn F
+GLIBC_2.33 _IO_do_write F
+GLIBC_2.33 _IO_doallocbuf F
+GLIBC_2.33 _IO_fclose F
+GLIBC_2.33 _IO_fdopen F
+GLIBC_2.33 _IO_feof F
+GLIBC_2.33 _IO_ferror F
+GLIBC_2.33 _IO_fflush F
+GLIBC_2.33 _IO_fgetpos F
+GLIBC_2.33 _IO_fgetpos64 F
+GLIBC_2.33 _IO_fgets F
+GLIBC_2.33 _IO_file_attach F
+GLIBC_2.33 _IO_file_close F
+GLIBC_2.33 _IO_file_close_it F
+GLIBC_2.33 _IO_file_doallocate F
+GLIBC_2.33 _IO_file_finish F
+GLIBC_2.33 _IO_file_fopen F
+GLIBC_2.33 _IO_file_init F
+GLIBC_2.33 _IO_file_jumps D 0x54
+GLIBC_2.33 _IO_file_open F
+GLIBC_2.33 _IO_file_overflow F
+GLIBC_2.33 _IO_file_read F
+GLIBC_2.33 _IO_file_seek F
+GLIBC_2.33 _IO_file_seekoff F
+GLIBC_2.33 _IO_file_setbuf F
+GLIBC_2.33 _IO_file_stat F
+GLIBC_2.33 _IO_file_sync F
+GLIBC_2.33 _IO_file_underflow F
+GLIBC_2.33 _IO_file_write F
+GLIBC_2.33 _IO_file_xsputn F
+GLIBC_2.33 _IO_flockfile F
+GLIBC_2.33 _IO_flush_all F
+GLIBC_2.33 _IO_flush_all_linebuffered F
+GLIBC_2.33 _IO_fopen F
+GLIBC_2.33 _IO_fprintf F
+GLIBC_2.33 _IO_fputs F
+GLIBC_2.33 _IO_fread F
+GLIBC_2.33 _IO_free_backup_area F
+GLIBC_2.33 _IO_free_wbackup_area F
+GLIBC_2.33 _IO_fsetpos F
+GLIBC_2.33 _IO_fsetpos64 F
+GLIBC_2.33 _IO_ftell F
+GLIBC_2.33 _IO_ftrylockfile F
+GLIBC_2.33 _IO_funlockfile F
+GLIBC_2.33 _IO_fwrite F
+GLIBC_2.33 _IO_getc F
+GLIBC_2.33 _IO_getline F
+GLIBC_2.33 _IO_getline_info F
+GLIBC_2.33 _IO_gets F
+GLIBC_2.33 _IO_init F
+GLIBC_2.33 _IO_init_marker F
+GLIBC_2.33 _IO_init_wmarker F
+GLIBC_2.33 _IO_iter_begin F
+GLIBC_2.33 _IO_iter_end F
+GLIBC_2.33 _IO_iter_file F
+GLIBC_2.33 _IO_iter_next F
+GLIBC_2.33 _IO_least_wmarker F
+GLIBC_2.33 _IO_link_in F
+GLIBC_2.33 _IO_list_all D 0x4
+GLIBC_2.33 _IO_list_lock F
+GLIBC_2.33 _IO_list_resetlock F
+GLIBC_2.33 _IO_list_unlock F
+GLIBC_2.33 _IO_marker_delta F
+GLIBC_2.33 _IO_marker_difference F
+GLIBC_2.33 _IO_padn F
+GLIBC_2.33 _IO_peekc_locked F
+GLIBC_2.33 _IO_popen F
+GLIBC_2.33 _IO_printf F
+GLIBC_2.33 _IO_proc_close F
+GLIBC_2.33 _IO_proc_open F
+GLIBC_2.33 _IO_putc F
+GLIBC_2.33 _IO_puts F
+GLIBC_2.33 _IO_remove_marker F
+GLIBC_2.33 _IO_seekmark F
+GLIBC_2.33 _IO_seekoff F
+GLIBC_2.33 _IO_seekpos F
+GLIBC_2.33 _IO_seekwmark F
+GLIBC_2.33 _IO_setb F
+GLIBC_2.33 _IO_setbuffer F
+GLIBC_2.33 _IO_setvbuf F
+GLIBC_2.33 _IO_sgetn F
+GLIBC_2.33 _IO_sprintf F
+GLIBC_2.33 _IO_sputbackc F
+GLIBC_2.33 _IO_sputbackwc F
+GLIBC_2.33 _IO_sscanf F
+GLIBC_2.33 _IO_str_init_readonly F
+GLIBC_2.33 _IO_str_init_static F
+GLIBC_2.33 _IO_str_overflow F
+GLIBC_2.33 _IO_str_pbackfail F
+GLIBC_2.33 _IO_str_seekoff F
+GLIBC_2.33 _IO_str_underflow F
+GLIBC_2.33 _IO_sungetc F
+GLIBC_2.33 _IO_sungetwc F
+GLIBC_2.33 _IO_switch_to_get_mode F
+GLIBC_2.33 _IO_switch_to_main_wget_area F
+GLIBC_2.33 _IO_switch_to_wbackup_area F
+GLIBC_2.33 _IO_switch_to_wget_mode F
+GLIBC_2.33 _IO_un_link F
+GLIBC_2.33 _IO_ungetc F
+GLIBC_2.33 _IO_unsave_markers F
+GLIBC_2.33 _IO_unsave_wmarkers F
+GLIBC_2.33 _IO_vfprintf F
+GLIBC_2.33 _IO_vsprintf F
+GLIBC_2.33 _IO_wdefault_doallocate F
+GLIBC_2.33 _IO_wdefault_finish F
+GLIBC_2.33 _IO_wdefault_pbackfail F
+GLIBC_2.33 _IO_wdefault_uflow F
+GLIBC_2.33 _IO_wdefault_xsgetn F
+GLIBC_2.33 _IO_wdefault_xsputn F
+GLIBC_2.33 _IO_wdo_write F
+GLIBC_2.33 _IO_wdoallocbuf F
+GLIBC_2.33 _IO_wfile_jumps D 0x54
+GLIBC_2.33 _IO_wfile_overflow F
+GLIBC_2.33 _IO_wfile_seekoff F
+GLIBC_2.33 _IO_wfile_sync F
+GLIBC_2.33 _IO_wfile_underflow F
+GLIBC_2.33 _IO_wfile_xsputn F
+GLIBC_2.33 _IO_wmarker_delta F
+GLIBC_2.33 _IO_wsetb F
+GLIBC_2.33 ___brk_addr D 0x4
+GLIBC_2.33 __adjtimex F
+GLIBC_2.33 __after_morecore_hook D 0x4
+GLIBC_2.33 __argz_count F
+GLIBC_2.33 __argz_next F
+GLIBC_2.33 __argz_stringify F
+GLIBC_2.33 __asprintf F
+GLIBC_2.33 __asprintf_chk F
+GLIBC_2.33 __assert F
+GLIBC_2.33 __assert_fail F
+GLIBC_2.33 __assert_perror_fail F
+GLIBC_2.33 __backtrace F
+GLIBC_2.33 __backtrace_symbols F
+GLIBC_2.33 __backtrace_symbols_fd F
+GLIBC_2.33 __bsd_getpgrp F
+GLIBC_2.33 __bzero F
+GLIBC_2.33 __check_rhosts_file D 0x4
+GLIBC_2.33 __chk_fail F
+GLIBC_2.33 __clone F
+GLIBC_2.33 __close F
+GLIBC_2.33 __cmsg_nxthdr F
+GLIBC_2.33 __confstr_chk F
+GLIBC_2.33 __connect F
+GLIBC_2.33 __ctype_b_loc F
+GLIBC_2.33 __ctype_get_mb_cur_max F
+GLIBC_2.33 __ctype_tolower_loc F
+GLIBC_2.33 __ctype_toupper_loc F
+GLIBC_2.33 __curbrk D 0x4
+GLIBC_2.33 __cxa_at_quick_exit F
+GLIBC_2.33 __cxa_atexit F
+GLIBC_2.33 __cxa_finalize F
+GLIBC_2.33 __cxa_thread_atexit_impl F
+GLIBC_2.33 __cyg_profile_func_enter F
+GLIBC_2.33 __cyg_profile_func_exit F
+GLIBC_2.33 __daylight D 0x4
+GLIBC_2.33 __dcgettext F
+GLIBC_2.33 __default_morecore F
+GLIBC_2.33 __dgettext F
+GLIBC_2.33 __dprintf_chk F
+GLIBC_2.33 __dup2 F
+GLIBC_2.33 __duplocale F
+GLIBC_2.33 __endmntent F
+GLIBC_2.33 __environ D 0x4
+GLIBC_2.33 __errno_location F
+GLIBC_2.33 __explicit_bzero_chk F
+GLIBC_2.33 __fbufsize F
+GLIBC_2.33 __fcntl F
+GLIBC_2.33 __fdelt_chk F
+GLIBC_2.33 __fdelt_warn F
+GLIBC_2.33 __ffs F
+GLIBC_2.33 __fgets_chk F
+GLIBC_2.33 __fgets_unlocked_chk F
+GLIBC_2.33 __fgetws_chk F
+GLIBC_2.33 __fgetws_unlocked_chk F
+GLIBC_2.33 __finite F
+GLIBC_2.33 __finitef F
+GLIBC_2.33 __finitel F
+GLIBC_2.33 __flbf F
+GLIBC_2.33 __fork F
+GLIBC_2.33 __fpending F
+GLIBC_2.33 __fprintf_chk F
+GLIBC_2.33 __fpu_control D 0x4
+GLIBC_2.33 __fpurge F
+GLIBC_2.33 __fread_chk F
+GLIBC_2.33 __fread_unlocked_chk F
+GLIBC_2.33 __freadable F
+GLIBC_2.33 __freading F
+GLIBC_2.33 __free_hook D 0x4
+GLIBC_2.33 __freelocale F
+GLIBC_2.33 __fsetlocking F
+GLIBC_2.33 __fwprintf_chk F
+GLIBC_2.33 __fwritable F
+GLIBC_2.33 __fwriting F
+GLIBC_2.33 __fxstat F
+GLIBC_2.33 __fxstat64 F
+GLIBC_2.33 __fxstatat F
+GLIBC_2.33 __fxstatat64 F
+GLIBC_2.33 __getauxval F
+GLIBC_2.33 __getcwd_chk F
+GLIBC_2.33 __getdelim F
+GLIBC_2.33 __getdomainname_chk F
+GLIBC_2.33 __getgroups_chk F
+GLIBC_2.33 __gethostname_chk F
+GLIBC_2.33 __getlogin_r_chk F
+GLIBC_2.33 __getmntent_r F
+GLIBC_2.33 __getpagesize F
+GLIBC_2.33 __getpgid F
+GLIBC_2.33 __getpid F
+GLIBC_2.33 __gets_chk F
+GLIBC_2.33 __gettimeofday F
+GLIBC_2.33 __getwd_chk F
+GLIBC_2.33 __gmtime_r F
+GLIBC_2.33 __h_errno_location F
+GLIBC_2.33 __isalnum_l F
+GLIBC_2.33 __isalpha_l F
+GLIBC_2.33 __isascii_l F
+GLIBC_2.33 __isblank_l F
+GLIBC_2.33 __iscntrl_l F
+GLIBC_2.33 __isctype F
+GLIBC_2.33 __isdigit_l F
+GLIBC_2.33 __isgraph_l F
+GLIBC_2.33 __isinf F
+GLIBC_2.33 __isinff F
+GLIBC_2.33 __isinfl F
+GLIBC_2.33 __islower_l F
+GLIBC_2.33 __isnan F
+GLIBC_2.33 __isnanf F
+GLIBC_2.33 __isnanl F
+GLIBC_2.33 __isoc99_fscanf F
+GLIBC_2.33 __isoc99_fwscanf F
+GLIBC_2.33 __isoc99_scanf F
+GLIBC_2.33 __isoc99_sscanf F
+GLIBC_2.33 __isoc99_swscanf F
+GLIBC_2.33 __isoc99_vfscanf F
+GLIBC_2.33 __isoc99_vfwscanf F
+GLIBC_2.33 __isoc99_vscanf F
+GLIBC_2.33 __isoc99_vsscanf F
+GLIBC_2.33 __isoc99_vswscanf F
+GLIBC_2.33 __isoc99_vwscanf F
+GLIBC_2.33 __isoc99_wscanf F
+GLIBC_2.33 __isprint_l F
+GLIBC_2.33 __ispunct_l F
+GLIBC_2.33 __isspace_l F
+GLIBC_2.33 __isupper_l F
+GLIBC_2.33 __iswalnum_l F
+GLIBC_2.33 __iswalpha_l F
+GLIBC_2.33 __iswblank_l F
+GLIBC_2.33 __iswcntrl_l F
+GLIBC_2.33 __iswctype F
+GLIBC_2.33 __iswctype_l F
+GLIBC_2.33 __iswdigit_l F
+GLIBC_2.33 __iswgraph_l F
+GLIBC_2.33 __iswlower_l F
+GLIBC_2.33 __iswprint_l F
+GLIBC_2.33 __iswpunct_l F
+GLIBC_2.33 __iswspace_l F
+GLIBC_2.33 __iswupper_l F
+GLIBC_2.33 __iswxdigit_l F
+GLIBC_2.33 __isxdigit_l F
+GLIBC_2.33 __ivaliduser F
+GLIBC_2.33 __libc_allocate_rtsig F
+GLIBC_2.33 __libc_calloc F
+GLIBC_2.33 __libc_current_sigrtmax F
+GLIBC_2.33 __libc_current_sigrtmin F
+GLIBC_2.33 __libc_free F
+GLIBC_2.33 __libc_freeres F
+GLIBC_2.33 __libc_init_first F
+GLIBC_2.33 __libc_mallinfo F
+GLIBC_2.33 __libc_malloc F
+GLIBC_2.33 __libc_mallopt F
+GLIBC_2.33 __libc_memalign F
+GLIBC_2.33 __libc_pvalloc F
+GLIBC_2.33 __libc_realloc F
+GLIBC_2.33 __libc_sa_len F
+GLIBC_2.33 __libc_single_threaded D 0x1
+GLIBC_2.33 __libc_start_main F
+GLIBC_2.33 __libc_valloc F
+GLIBC_2.33 __longjmp_chk F
+GLIBC_2.33 __lseek F
+GLIBC_2.33 __lxstat F
+GLIBC_2.33 __lxstat64 F
+GLIBC_2.33 __malloc_hook D 0x4
+GLIBC_2.33 __mbrlen F
+GLIBC_2.33 __mbrtowc F
+GLIBC_2.33 __mbsnrtowcs_chk F
+GLIBC_2.33 __mbsrtowcs_chk F
+GLIBC_2.33 __mbstowcs_chk F
+GLIBC_2.33 __memalign_hook D 0x4
+GLIBC_2.33 __memcpy_chk F
+GLIBC_2.33 __memmove_chk F
+GLIBC_2.33 __mempcpy F
+GLIBC_2.33 __mempcpy_chk F
+GLIBC_2.33 __memset_chk F
+GLIBC_2.33 __monstartup F
+GLIBC_2.33 __morecore D 0x4
+GLIBC_2.33 __nanosleep F
+GLIBC_2.33 __newlocale F
+GLIBC_2.33 __nl_langinfo_l F
+GLIBC_2.33 __nss_configure_lookup F
+GLIBC_2.33 __nss_hostname_digits_dots F
+GLIBC_2.33 __obstack_printf_chk F
+GLIBC_2.33 __obstack_vprintf_chk F
+GLIBC_2.33 __open F
+GLIBC_2.33 __open64 F
+GLIBC_2.33 __open64_2 F
+GLIBC_2.33 __open_2 F
+GLIBC_2.33 __openat64_2 F
+GLIBC_2.33 __openat_2 F
+GLIBC_2.33 __overflow F
+GLIBC_2.33 __pipe F
+GLIBC_2.33 __poll F
+GLIBC_2.33 __poll_chk F
+GLIBC_2.33 __posix_getopt F
+GLIBC_2.33 __ppoll_chk F
+GLIBC_2.33 __pread64 F
+GLIBC_2.33 __pread64_chk F
+GLIBC_2.33 __pread_chk F
+GLIBC_2.33 __printf_chk F
+GLIBC_2.33 __printf_fp F
+GLIBC_2.33 __profile_frequency F
+GLIBC_2.33 __progname D 0x4
+GLIBC_2.33 __progname_full D 0x4
+GLIBC_2.33 __ptsname_r_chk F
+GLIBC_2.33 __pwrite64 F
+GLIBC_2.33 __rawmemchr F
+GLIBC_2.33 __rcmd_errstr D 0x4
+GLIBC_2.33 __read F
+GLIBC_2.33 __read_chk F
+GLIBC_2.33 __readlink_chk F
+GLIBC_2.33 __readlinkat_chk F
+GLIBC_2.33 __realloc_hook D 0x4
+GLIBC_2.33 __realpath_chk F
+GLIBC_2.33 __recv_chk F
+GLIBC_2.33 __recvfrom_chk F
+GLIBC_2.33 __register_atfork F
+GLIBC_2.33 __res_init F
+GLIBC_2.33 __res_nclose F
+GLIBC_2.33 __res_ninit F
+GLIBC_2.33 __res_randomid F
+GLIBC_2.33 __res_state F
+GLIBC_2.33 __riscv_flush_icache F
+GLIBC_2.33 __sbrk F
+GLIBC_2.33 __sched_cpualloc F
+GLIBC_2.33 __sched_cpucount F
+GLIBC_2.33 __sched_cpufree F
+GLIBC_2.33 __sched_get_priority_max F
+GLIBC_2.33 __sched_get_priority_min F
+GLIBC_2.33 __sched_getparam F
+GLIBC_2.33 __sched_getscheduler F
+GLIBC_2.33 __sched_setscheduler F
+GLIBC_2.33 __sched_yield F
+GLIBC_2.33 __select F
+GLIBC_2.33 __send F
+GLIBC_2.33 __setmntent F
+GLIBC_2.33 __setpgid F
+GLIBC_2.33 __sigaction F
+GLIBC_2.33 __signbit F
+GLIBC_2.33 __signbitf F
+GLIBC_2.33 __signbitl F
+GLIBC_2.33 __sigpause F
+GLIBC_2.33 __sigsetjmp F
+GLIBC_2.33 __sigsuspend F
+GLIBC_2.33 __snprintf_chk F
+GLIBC_2.33 __sprintf_chk F
+GLIBC_2.33 __stack_chk_fail F
+GLIBC_2.33 __statfs F
+GLIBC_2.33 __stpcpy F
+GLIBC_2.33 __stpcpy_chk F
+GLIBC_2.33 __stpncpy F
+GLIBC_2.33 __stpncpy_chk F
+GLIBC_2.33 __strcasecmp F
+GLIBC_2.33 __strcasecmp_l F
+GLIBC_2.33 __strcasestr F
+GLIBC_2.33 __strcat_chk F
+GLIBC_2.33 __strcoll_l F
+GLIBC_2.33 __strcpy_chk F
+GLIBC_2.33 __strdup F
+GLIBC_2.33 __strerror_r F
+GLIBC_2.33 __strfmon_l F
+GLIBC_2.33 __strftime_l F
+GLIBC_2.33 __strncasecmp_l F
+GLIBC_2.33 __strncat_chk F
+GLIBC_2.33 __strncpy_chk F
+GLIBC_2.33 __strndup F
+GLIBC_2.33 __strsep_g F
+GLIBC_2.33 __strtod_internal F
+GLIBC_2.33 __strtod_l F
+GLIBC_2.33 __strtof_internal F
+GLIBC_2.33 __strtof_l F
+GLIBC_2.33 __strtok_r F
+GLIBC_2.33 __strtol_internal F
+GLIBC_2.33 __strtol_l F
+GLIBC_2.33 __strtold_internal F
+GLIBC_2.33 __strtold_l F
+GLIBC_2.33 __strtoll_internal F
+GLIBC_2.33 __strtoll_l F
+GLIBC_2.33 __strtoul_internal F
+GLIBC_2.33 __strtoul_l F
+GLIBC_2.33 __strtoull_internal F
+GLIBC_2.33 __strtoull_l F
+GLIBC_2.33 __strverscmp F
+GLIBC_2.33 __strxfrm_l F
+GLIBC_2.33 __swprintf_chk F
+GLIBC_2.33 __sysconf F
+GLIBC_2.33 __syslog_chk F
+GLIBC_2.33 __sysv_signal F
+GLIBC_2.33 __timezone D 0x4
+GLIBC_2.33 __toascii_l F
+GLIBC_2.33 __tolower_l F
+GLIBC_2.33 __toupper_l F
+GLIBC_2.33 __towctrans F
+GLIBC_2.33 __towctrans_l F
+GLIBC_2.33 __towlower_l F
+GLIBC_2.33 __towupper_l F
+GLIBC_2.33 __ttyname_r_chk F
+GLIBC_2.33 __tzname D 0x8
+GLIBC_2.33 __uflow F
+GLIBC_2.33 __underflow F
+GLIBC_2.33 __uselocale F
+GLIBC_2.33 __vasprintf_chk F
+GLIBC_2.33 __vdprintf_chk F
+GLIBC_2.33 __vfork F
+GLIBC_2.33 __vfprintf_chk F
+GLIBC_2.33 __vfscanf F
+GLIBC_2.33 __vfwprintf_chk F
+GLIBC_2.33 __vprintf_chk F
+GLIBC_2.33 __vsnprintf F
+GLIBC_2.33 __vsnprintf_chk F
+GLIBC_2.33 __vsprintf_chk F
+GLIBC_2.33 __vsscanf F
+GLIBC_2.33 __vswprintf_chk F
+GLIBC_2.33 __vsyslog_chk F
+GLIBC_2.33 __vwprintf_chk F
+GLIBC_2.33 __wait F
+GLIBC_2.33 __waitpid F
+GLIBC_2.33 __wcpcpy_chk F
+GLIBC_2.33 __wcpncpy_chk F
+GLIBC_2.33 __wcrtomb_chk F
+GLIBC_2.33 __wcscasecmp_l F
+GLIBC_2.33 __wcscat_chk F
+GLIBC_2.33 __wcscoll_l F
+GLIBC_2.33 __wcscpy_chk F
+GLIBC_2.33 __wcsftime_l F
+GLIBC_2.33 __wcsncasecmp_l F
+GLIBC_2.33 __wcsncat_chk F
+GLIBC_2.33 __wcsncpy_chk F
+GLIBC_2.33 __wcsnrtombs_chk F
+GLIBC_2.33 __wcsrtombs_chk F
+GLIBC_2.33 __wcstod_internal F
+GLIBC_2.33 __wcstod_l F
+GLIBC_2.33 __wcstof_internal F
+GLIBC_2.33 __wcstof_l F
+GLIBC_2.33 __wcstol_internal F
+GLIBC_2.33 __wcstol_l F
+GLIBC_2.33 __wcstold_internal F
+GLIBC_2.33 __wcstold_l F
+GLIBC_2.33 __wcstoll_internal F
+GLIBC_2.33 __wcstoll_l F
+GLIBC_2.33 __wcstombs_chk F
+GLIBC_2.33 __wcstoul_internal F
+GLIBC_2.33 __wcstoul_l F
+GLIBC_2.33 __wcstoull_internal F
+GLIBC_2.33 __wcstoull_l F
+GLIBC_2.33 __wcsxfrm_l F
+GLIBC_2.33 __wctomb_chk F
+GLIBC_2.33 __wctrans_l F
+GLIBC_2.33 __wctype_l F
+GLIBC_2.33 __wmemcpy_chk F
+GLIBC_2.33 __wmemmove_chk F
+GLIBC_2.33 __wmempcpy_chk F
+GLIBC_2.33 __wmemset_chk F
+GLIBC_2.33 __woverflow F
+GLIBC_2.33 __wprintf_chk F
+GLIBC_2.33 __write F
+GLIBC_2.33 __wuflow F
+GLIBC_2.33 __wunderflow F
+GLIBC_2.33 __xmknod F
+GLIBC_2.33 __xmknodat F
+GLIBC_2.33 __xpg_basename F
+GLIBC_2.33 __xpg_sigpause F
+GLIBC_2.33 __xpg_strerror_r F
+GLIBC_2.33 __xstat F
+GLIBC_2.33 __xstat64 F
+GLIBC_2.33 _dl_mcount_wrapper F
+GLIBC_2.33 _dl_mcount_wrapper_check F
+GLIBC_2.33 _environ D 0x4
+GLIBC_2.33 _exit F
+GLIBC_2.33 _flushlbf F
+GLIBC_2.33 _libc_intl_domainname D 0x5
+GLIBC_2.33 _longjmp F
+GLIBC_2.33 _mcleanup F
+GLIBC_2.33 _mcount F
+GLIBC_2.33 _nl_default_dirname D 0x12
+GLIBC_2.33 _nl_domain_bindings D 0x4
+GLIBC_2.33 _nl_msg_cat_cntr D 0x4
+GLIBC_2.33 _obstack_allocated_p F
+GLIBC_2.33 _obstack_begin F
+GLIBC_2.33 _obstack_begin_1 F
+GLIBC_2.33 _obstack_free F
+GLIBC_2.33 _obstack_memory_used F
+GLIBC_2.33 _obstack_newchunk F
+GLIBC_2.33 _res D 0x200
+GLIBC_2.33 _res_hconf D 0x30
+GLIBC_2.33 _setjmp F
+GLIBC_2.33 _tolower F
+GLIBC_2.33 _toupper F
+GLIBC_2.33 a64l F
+GLIBC_2.33 abort F
+GLIBC_2.33 abs F
+GLIBC_2.33 accept F
+GLIBC_2.33 accept4 F
+GLIBC_2.33 access F
+GLIBC_2.33 acct F
+GLIBC_2.33 addmntent F
+GLIBC_2.33 addseverity F
+GLIBC_2.33 adjtime F
+GLIBC_2.33 adjtimex F
+GLIBC_2.33 alarm F
+GLIBC_2.33 aligned_alloc F
+GLIBC_2.33 alphasort F
+GLIBC_2.33 alphasort64 F
+GLIBC_2.33 argp_err_exit_status D 0x4
+GLIBC_2.33 argp_error F
+GLIBC_2.33 argp_failure F
+GLIBC_2.33 argp_help F
+GLIBC_2.33 argp_parse F
+GLIBC_2.33 argp_program_bug_address D 0x4
+GLIBC_2.33 argp_program_version D 0x4
+GLIBC_2.33 argp_program_version_hook D 0x4
+GLIBC_2.33 argp_state_help F
+GLIBC_2.33 argp_usage F
+GLIBC_2.33 argz_add F
+GLIBC_2.33 argz_add_sep F
+GLIBC_2.33 argz_append F
+GLIBC_2.33 argz_count F
+GLIBC_2.33 argz_create F
+GLIBC_2.33 argz_create_sep F
+GLIBC_2.33 argz_delete F
+GLIBC_2.33 argz_extract F
+GLIBC_2.33 argz_insert F
+GLIBC_2.33 argz_next F
+GLIBC_2.33 argz_replace F
+GLIBC_2.33 argz_stringify F
+GLIBC_2.33 asctime F
+GLIBC_2.33 asctime_r F
+GLIBC_2.33 asprintf F
+GLIBC_2.33 atof F
+GLIBC_2.33 atoi F
+GLIBC_2.33 atol F
+GLIBC_2.33 atoll F
+GLIBC_2.33 backtrace F
+GLIBC_2.33 backtrace_symbols F
+GLIBC_2.33 backtrace_symbols_fd F
+GLIBC_2.33 basename F
+GLIBC_2.33 bcmp F
+GLIBC_2.33 bcopy F
+GLIBC_2.33 bind F
+GLIBC_2.33 bind_textdomain_codeset F
+GLIBC_2.33 bindresvport F
+GLIBC_2.33 bindtextdomain F
+GLIBC_2.33 brk F
+GLIBC_2.33 bsd_signal F
+GLIBC_2.33 bsearch F
+GLIBC_2.33 btowc F
+GLIBC_2.33 bzero F
+GLIBC_2.33 c16rtomb F
+GLIBC_2.33 c32rtomb F
+GLIBC_2.33 calloc F
+GLIBC_2.33 canonicalize_file_name F
+GLIBC_2.33 capget F
+GLIBC_2.33 capset F
+GLIBC_2.33 catclose F
+GLIBC_2.33 catgets F
+GLIBC_2.33 catopen F
+GLIBC_2.33 cfgetispeed F
+GLIBC_2.33 cfgetospeed F
+GLIBC_2.33 cfmakeraw F
+GLIBC_2.33 cfsetispeed F
+GLIBC_2.33 cfsetospeed F
+GLIBC_2.33 cfsetspeed F
+GLIBC_2.33 chdir F
+GLIBC_2.33 chflags F
+GLIBC_2.33 chmod F
+GLIBC_2.33 chown F
+GLIBC_2.33 chroot F
+GLIBC_2.33 clearenv F
+GLIBC_2.33 clearerr F
+GLIBC_2.33 clearerr_unlocked F
+GLIBC_2.33 clock F
+GLIBC_2.33 clock_adjtime F
+GLIBC_2.33 clock_getcpuclockid F
+GLIBC_2.33 clock_getres F
+GLIBC_2.33 clock_gettime F
+GLIBC_2.33 clock_nanosleep F
+GLIBC_2.33 clock_settime F
+GLIBC_2.33 clone F
+GLIBC_2.33 close F
+GLIBC_2.33 closedir F
+GLIBC_2.33 closelog F
+GLIBC_2.33 confstr F
+GLIBC_2.33 connect F
+GLIBC_2.33 copy_file_range F
+GLIBC_2.33 copysign F
+GLIBC_2.33 copysignf F
+GLIBC_2.33 copysignl F
+GLIBC_2.33 creat F
+GLIBC_2.33 creat64 F
+GLIBC_2.33 ctermid F
+GLIBC_2.33 ctime F
+GLIBC_2.33 ctime_r F
+GLIBC_2.33 cuserid F
+GLIBC_2.33 daemon F
+GLIBC_2.33 daylight D 0x4
+GLIBC_2.33 dcgettext F
+GLIBC_2.33 dcngettext F
+GLIBC_2.33 delete_module F
+GLIBC_2.33 dgettext F
+GLIBC_2.33 difftime F
+GLIBC_2.33 dirfd F
+GLIBC_2.33 dirname F
+GLIBC_2.33 div F
+GLIBC_2.33 dl_iterate_phdr F
+GLIBC_2.33 dngettext F
+GLIBC_2.33 dprintf F
+GLIBC_2.33 drand48 F
+GLIBC_2.33 drand48_r F
+GLIBC_2.33 dup F
+GLIBC_2.33 dup2 F
+GLIBC_2.33 dup3 F
+GLIBC_2.33 duplocale F
+GLIBC_2.33 dysize F
+GLIBC_2.33 eaccess F
+GLIBC_2.33 ecvt F
+GLIBC_2.33 ecvt_r F
+GLIBC_2.33 endaliasent F
+GLIBC_2.33 endfsent F
+GLIBC_2.33 endgrent F
+GLIBC_2.33 endhostent F
+GLIBC_2.33 endmntent F
+GLIBC_2.33 endnetent F
+GLIBC_2.33 endnetgrent F
+GLIBC_2.33 endprotoent F
+GLIBC_2.33 endpwent F
+GLIBC_2.33 endrpcent F
+GLIBC_2.33 endservent F
+GLIBC_2.33 endsgent F
+GLIBC_2.33 endspent F
+GLIBC_2.33 endttyent F
+GLIBC_2.33 endusershell F
+GLIBC_2.33 endutent F
+GLIBC_2.33 endutxent F
+GLIBC_2.33 environ D 0x4
+GLIBC_2.33 envz_add F
+GLIBC_2.33 envz_entry F
+GLIBC_2.33 envz_get F
+GLIBC_2.33 envz_merge F
+GLIBC_2.33 envz_remove F
+GLIBC_2.33 envz_strip F
+GLIBC_2.33 epoll_create F
+GLIBC_2.33 epoll_create1 F
+GLIBC_2.33 epoll_ctl F
+GLIBC_2.33 epoll_pwait F
+GLIBC_2.33 epoll_wait F
+GLIBC_2.33 erand48 F
+GLIBC_2.33 erand48_r F
+GLIBC_2.33 err F
+GLIBC_2.33 error F
+GLIBC_2.33 error_at_line F
+GLIBC_2.33 error_message_count D 0x4
+GLIBC_2.33 error_one_per_line D 0x4
+GLIBC_2.33 error_print_progname D 0x4
+GLIBC_2.33 errx F
+GLIBC_2.33 ether_aton F
+GLIBC_2.33 ether_aton_r F
+GLIBC_2.33 ether_hostton F
+GLIBC_2.33 ether_line F
+GLIBC_2.33 ether_ntoa F
+GLIBC_2.33 ether_ntoa_r F
+GLIBC_2.33 ether_ntohost F
+GLIBC_2.33 euidaccess F
+GLIBC_2.33 eventfd F
+GLIBC_2.33 eventfd_read F
+GLIBC_2.33 eventfd_write F
+GLIBC_2.33 execl F
+GLIBC_2.33 execle F
+GLIBC_2.33 execlp F
+GLIBC_2.33 execv F
+GLIBC_2.33 execve F
+GLIBC_2.33 execvp F
+GLIBC_2.33 execvpe F
+GLIBC_2.33 exit F
+GLIBC_2.33 explicit_bzero F
+GLIBC_2.33 faccessat F
+GLIBC_2.33 fallocate F
+GLIBC_2.33 fallocate64 F
+GLIBC_2.33 fanotify_init F
+GLIBC_2.33 fanotify_mark F
+GLIBC_2.33 fchdir F
+GLIBC_2.33 fchflags F
+GLIBC_2.33 fchmod F
+GLIBC_2.33 fchmodat F
+GLIBC_2.33 fchown F
+GLIBC_2.33 fchownat F
+GLIBC_2.33 fclose F
+GLIBC_2.33 fcloseall F
+GLIBC_2.33 fcntl F
+GLIBC_2.33 fcntl64 F
+GLIBC_2.33 fcvt F
+GLIBC_2.33 fcvt_r F
+GLIBC_2.33 fdatasync F
+GLIBC_2.33 fdopen F
+GLIBC_2.33 fdopendir F
+GLIBC_2.33 feof F
+GLIBC_2.33 feof_unlocked F
+GLIBC_2.33 ferror F
+GLIBC_2.33 ferror_unlocked F
+GLIBC_2.33 fexecve F
+GLIBC_2.33 fflush F
+GLIBC_2.33 fflush_unlocked F
+GLIBC_2.33 ffs F
+GLIBC_2.33 ffsl F
+GLIBC_2.33 ffsll F
+GLIBC_2.33 fgetc F
+GLIBC_2.33 fgetc_unlocked F
+GLIBC_2.33 fgetgrent F
+GLIBC_2.33 fgetgrent_r F
+GLIBC_2.33 fgetpos F
+GLIBC_2.33 fgetpos64 F
+GLIBC_2.33 fgetpwent F
+GLIBC_2.33 fgetpwent_r F
+GLIBC_2.33 fgets F
+GLIBC_2.33 fgets_unlocked F
+GLIBC_2.33 fgetsgent F
+GLIBC_2.33 fgetsgent_r F
+GLIBC_2.33 fgetspent F
+GLIBC_2.33 fgetspent_r F
+GLIBC_2.33 fgetwc F
+GLIBC_2.33 fgetwc_unlocked F
+GLIBC_2.33 fgetws F
+GLIBC_2.33 fgetws_unlocked F
+GLIBC_2.33 fgetxattr F
+GLIBC_2.33 fileno F
+GLIBC_2.33 fileno_unlocked F
+GLIBC_2.33 finite F
+GLIBC_2.33 finitef F
+GLIBC_2.33 finitel F
+GLIBC_2.33 flistxattr F
+GLIBC_2.33 flock F
+GLIBC_2.33 flockfile F
+GLIBC_2.33 fmemopen F
+GLIBC_2.33 fmtmsg F
+GLIBC_2.33 fnmatch F
+GLIBC_2.33 fopen F
+GLIBC_2.33 fopen64 F
+GLIBC_2.33 fopencookie F
+GLIBC_2.33 fork F
+GLIBC_2.33 fpathconf F
+GLIBC_2.33 fprintf F
+GLIBC_2.33 fputc F
+GLIBC_2.33 fputc_unlocked F
+GLIBC_2.33 fputs F
+GLIBC_2.33 fputs_unlocked F
+GLIBC_2.33 fputwc F
+GLIBC_2.33 fputwc_unlocked F
+GLIBC_2.33 fputws F
+GLIBC_2.33 fputws_unlocked F
+GLIBC_2.33 fread F
+GLIBC_2.33 fread_unlocked F
+GLIBC_2.33 free F
+GLIBC_2.33 freeaddrinfo F
+GLIBC_2.33 freeifaddrs F
+GLIBC_2.33 freelocale F
+GLIBC_2.33 fremovexattr F
+GLIBC_2.33 freopen F
+GLIBC_2.33 freopen64 F
+GLIBC_2.33 frexp F
+GLIBC_2.33 frexpf F
+GLIBC_2.33 frexpl F
+GLIBC_2.33 fscanf F
+GLIBC_2.33 fseek F
+GLIBC_2.33 fseeko F
+GLIBC_2.33 fseeko64 F
+GLIBC_2.33 fsetpos F
+GLIBC_2.33 fsetpos64 F
+GLIBC_2.33 fsetxattr F
+GLIBC_2.33 fstatfs F
+GLIBC_2.33 fstatfs64 F
+GLIBC_2.33 fstatvfs F
+GLIBC_2.33 fstatvfs64 F
+GLIBC_2.33 fsync F
+GLIBC_2.33 ftell F
+GLIBC_2.33 ftello F
+GLIBC_2.33 ftello64 F
+GLIBC_2.33 ftime F
+GLIBC_2.33 ftok F
+GLIBC_2.33 ftruncate F
+GLIBC_2.33 ftruncate64 F
+GLIBC_2.33 ftrylockfile F
+GLIBC_2.33 fts64_children F
+GLIBC_2.33 fts64_close F
+GLIBC_2.33 fts64_open F
+GLIBC_2.33 fts64_read F
+GLIBC_2.33 fts64_set F
+GLIBC_2.33 fts_children F
+GLIBC_2.33 fts_close F
+GLIBC_2.33 fts_open F
+GLIBC_2.33 fts_read F
+GLIBC_2.33 fts_set F
+GLIBC_2.33 ftw F
+GLIBC_2.33 ftw64 F
+GLIBC_2.33 funlockfile F
+GLIBC_2.33 futimens F
+GLIBC_2.33 futimes F
+GLIBC_2.33 futimesat F
+GLIBC_2.33 fwide F
+GLIBC_2.33 fwprintf F
+GLIBC_2.33 fwrite F
+GLIBC_2.33 fwrite_unlocked F
+GLIBC_2.33 fwscanf F
+GLIBC_2.33 gai_strerror F
+GLIBC_2.33 gcvt F
+GLIBC_2.33 get_avphys_pages F
+GLIBC_2.33 get_current_dir_name F
+GLIBC_2.33 get_nprocs F
+GLIBC_2.33 get_nprocs_conf F
+GLIBC_2.33 get_phys_pages F
+GLIBC_2.33 getaddrinfo F
+GLIBC_2.33 getaliasbyname F
+GLIBC_2.33 getaliasbyname_r F
+GLIBC_2.33 getaliasent F
+GLIBC_2.33 getaliasent_r F
+GLIBC_2.33 getauxval F
+GLIBC_2.33 getc F
+GLIBC_2.33 getc_unlocked F
+GLIBC_2.33 getchar F
+GLIBC_2.33 getchar_unlocked F
+GLIBC_2.33 getcontext F
+GLIBC_2.33 getcpu F
+GLIBC_2.33 getcwd F
+GLIBC_2.33 getdate F
+GLIBC_2.33 getdate_err D 0x4
+GLIBC_2.33 getdate_r F
+GLIBC_2.33 getdelim F
+GLIBC_2.33 getdents64 F
+GLIBC_2.33 getdirentries F
+GLIBC_2.33 getdirentries64 F
+GLIBC_2.33 getdomainname F
+GLIBC_2.33 getdtablesize F
+GLIBC_2.33 getegid F
+GLIBC_2.33 getentropy F
+GLIBC_2.33 getenv F
+GLIBC_2.33 geteuid F
+GLIBC_2.33 getfsent F
+GLIBC_2.33 getfsfile F
+GLIBC_2.33 getfsspec F
+GLIBC_2.33 getgid F
+GLIBC_2.33 getgrent F
+GLIBC_2.33 getgrent_r F
+GLIBC_2.33 getgrgid F
+GLIBC_2.33 getgrgid_r F
+GLIBC_2.33 getgrnam F
+GLIBC_2.33 getgrnam_r F
+GLIBC_2.33 getgrouplist F
+GLIBC_2.33 getgroups F
+GLIBC_2.33 gethostbyaddr F
+GLIBC_2.33 gethostbyaddr_r F
+GLIBC_2.33 gethostbyname F
+GLIBC_2.33 gethostbyname2 F
+GLIBC_2.33 gethostbyname2_r F
+GLIBC_2.33 gethostbyname_r F
+GLIBC_2.33 gethostent F
+GLIBC_2.33 gethostent_r F
+GLIBC_2.33 gethostid F
+GLIBC_2.33 gethostname F
+GLIBC_2.33 getifaddrs F
+GLIBC_2.33 getipv4sourcefilter F
+GLIBC_2.33 getitimer F
+GLIBC_2.33 getline F
+GLIBC_2.33 getloadavg F
+GLIBC_2.33 getlogin F
+GLIBC_2.33 getlogin_r F
+GLIBC_2.33 getmntent F
+GLIBC_2.33 getmntent_r F
+GLIBC_2.33 getnameinfo F
+GLIBC_2.33 getnetbyaddr F
+GLIBC_2.33 getnetbyaddr_r F
+GLIBC_2.33 getnetbyname F
+GLIBC_2.33 getnetbyname_r F
+GLIBC_2.33 getnetent F
+GLIBC_2.33 getnetent_r F
+GLIBC_2.33 getnetgrent F
+GLIBC_2.33 getnetgrent_r F
+GLIBC_2.33 getopt F
+GLIBC_2.33 getopt_long F
+GLIBC_2.33 getopt_long_only F
+GLIBC_2.33 getpagesize F
+GLIBC_2.33 getpass F
+GLIBC_2.33 getpeername F
+GLIBC_2.33 getpgid F
+GLIBC_2.33 getpgrp F
+GLIBC_2.33 getpid F
+GLIBC_2.33 getppid F
+GLIBC_2.33 getpriority F
+GLIBC_2.33 getprotobyname F
+GLIBC_2.33 getprotobyname_r F
+GLIBC_2.33 getprotobynumber F
+GLIBC_2.33 getprotobynumber_r F
+GLIBC_2.33 getprotoent F
+GLIBC_2.33 getprotoent_r F
+GLIBC_2.33 getpt F
+GLIBC_2.33 getpw F
+GLIBC_2.33 getpwent F
+GLIBC_2.33 getpwent_r F
+GLIBC_2.33 getpwnam F
+GLIBC_2.33 getpwnam_r F
+GLIBC_2.33 getpwuid F
+GLIBC_2.33 getpwuid_r F
+GLIBC_2.33 getrandom F
+GLIBC_2.33 getresgid F
+GLIBC_2.33 getresuid F
+GLIBC_2.33 getrlimit F
+GLIBC_2.33 getrlimit64 F
+GLIBC_2.33 getrpcbyname F
+GLIBC_2.33 getrpcbyname_r F
+GLIBC_2.33 getrpcbynumber F
+GLIBC_2.33 getrpcbynumber_r F
+GLIBC_2.33 getrpcent F
+GLIBC_2.33 getrpcent_r F
+GLIBC_2.33 getrusage F
+GLIBC_2.33 gets F
+GLIBC_2.33 getservbyname F
+GLIBC_2.33 getservbyname_r F
+GLIBC_2.33 getservbyport F
+GLIBC_2.33 getservbyport_r F
+GLIBC_2.33 getservent F
+GLIBC_2.33 getservent_r F
+GLIBC_2.33 getsgent F
+GLIBC_2.33 getsgent_r F
+GLIBC_2.33 getsgnam F
+GLIBC_2.33 getsgnam_r F
+GLIBC_2.33 getsid F
+GLIBC_2.33 getsockname F
+GLIBC_2.33 getsockopt F
+GLIBC_2.33 getsourcefilter F
+GLIBC_2.33 getspent F
+GLIBC_2.33 getspent_r F
+GLIBC_2.33 getspnam F
+GLIBC_2.33 getspnam_r F
+GLIBC_2.33 getsubopt F
+GLIBC_2.33 gettext F
+GLIBC_2.33 gettid F
+GLIBC_2.33 gettimeofday F
+GLIBC_2.33 getttyent F
+GLIBC_2.33 getttynam F
+GLIBC_2.33 getuid F
+GLIBC_2.33 getusershell F
+GLIBC_2.33 getutent F
+GLIBC_2.33 getutent_r F
+GLIBC_2.33 getutid F
+GLIBC_2.33 getutid_r F
+GLIBC_2.33 getutline F
+GLIBC_2.33 getutline_r F
+GLIBC_2.33 getutmp F
+GLIBC_2.33 getutmpx F
+GLIBC_2.33 getutxent F
+GLIBC_2.33 getutxid F
+GLIBC_2.33 getutxline F
+GLIBC_2.33 getw F
+GLIBC_2.33 getwc F
+GLIBC_2.33 getwc_unlocked F
+GLIBC_2.33 getwchar F
+GLIBC_2.33 getwchar_unlocked F
+GLIBC_2.33 getwd F
+GLIBC_2.33 getxattr F
+GLIBC_2.33 glob F
+GLIBC_2.33 glob64 F
+GLIBC_2.33 glob_pattern_p F
+GLIBC_2.33 globfree F
+GLIBC_2.33 globfree64 F
+GLIBC_2.33 gmtime F
+GLIBC_2.33 gmtime_r F
+GLIBC_2.33 gnu_dev_major F
+GLIBC_2.33 gnu_dev_makedev F
+GLIBC_2.33 gnu_dev_minor F
+GLIBC_2.33 gnu_get_libc_release F
+GLIBC_2.33 gnu_get_libc_version F
+GLIBC_2.33 grantpt F
+GLIBC_2.33 group_member F
+GLIBC_2.33 gsignal F
+GLIBC_2.33 gtty F
+GLIBC_2.33 h_errlist D 0x14
+GLIBC_2.33 h_nerr D 0x4
+GLIBC_2.33 hasmntopt F
+GLIBC_2.33 hcreate F
+GLIBC_2.33 hcreate_r F
+GLIBC_2.33 hdestroy F
+GLIBC_2.33 hdestroy_r F
+GLIBC_2.33 herror F
+GLIBC_2.33 hsearch F
+GLIBC_2.33 hsearch_r F
+GLIBC_2.33 hstrerror F
+GLIBC_2.33 htonl F
+GLIBC_2.33 htons F
+GLIBC_2.33 iconv F
+GLIBC_2.33 iconv_close F
+GLIBC_2.33 iconv_open F
+GLIBC_2.33 if_freenameindex F
+GLIBC_2.33 if_indextoname F
+GLIBC_2.33 if_nameindex F
+GLIBC_2.33 if_nametoindex F
+GLIBC_2.33 imaxabs F
+GLIBC_2.33 imaxdiv F
+GLIBC_2.33 in6addr_any D 0x10
+GLIBC_2.33 in6addr_loopback D 0x10
+GLIBC_2.33 index F
+GLIBC_2.33 inet6_opt_append F
+GLIBC_2.33 inet6_opt_find F
+GLIBC_2.33 inet6_opt_finish F
+GLIBC_2.33 inet6_opt_get_val F
+GLIBC_2.33 inet6_opt_init F
+GLIBC_2.33 inet6_opt_next F
+GLIBC_2.33 inet6_opt_set_val F
+GLIBC_2.33 inet6_option_alloc F
+GLIBC_2.33 inet6_option_append F
+GLIBC_2.33 inet6_option_find F
+GLIBC_2.33 inet6_option_init F
+GLIBC_2.33 inet6_option_next F
+GLIBC_2.33 inet6_option_space F
+GLIBC_2.33 inet6_rth_add F
+GLIBC_2.33 inet6_rth_getaddr F
+GLIBC_2.33 inet6_rth_init F
+GLIBC_2.33 inet6_rth_reverse F
+GLIBC_2.33 inet6_rth_segments F
+GLIBC_2.33 inet6_rth_space F
+GLIBC_2.33 inet_addr F
+GLIBC_2.33 inet_aton F
+GLIBC_2.33 inet_lnaof F
+GLIBC_2.33 inet_makeaddr F
+GLIBC_2.33 inet_netof F
+GLIBC_2.33 inet_network F
+GLIBC_2.33 inet_nsap_addr F
+GLIBC_2.33 inet_nsap_ntoa F
+GLIBC_2.33 inet_ntoa F
+GLIBC_2.33 inet_ntop F
+GLIBC_2.33 inet_pton F
+GLIBC_2.33 init_module F
+GLIBC_2.33 initgroups F
+GLIBC_2.33 initstate F
+GLIBC_2.33 initstate_r F
+GLIBC_2.33 innetgr F
+GLIBC_2.33 inotify_add_watch F
+GLIBC_2.33 inotify_init F
+GLIBC_2.33 inotify_init1 F
+GLIBC_2.33 inotify_rm_watch F
+GLIBC_2.33 insque F
+GLIBC_2.33 ioctl F
+GLIBC_2.33 iruserok F
+GLIBC_2.33 iruserok_af F
+GLIBC_2.33 isalnum F
+GLIBC_2.33 isalnum_l F
+GLIBC_2.33 isalpha F
+GLIBC_2.33 isalpha_l F
+GLIBC_2.33 isascii F
+GLIBC_2.33 isatty F
+GLIBC_2.33 isblank F
+GLIBC_2.33 isblank_l F
+GLIBC_2.33 iscntrl F
+GLIBC_2.33 iscntrl_l F
+GLIBC_2.33 isctype F
+GLIBC_2.33 isdigit F
+GLIBC_2.33 isdigit_l F
+GLIBC_2.33 isfdtype F
+GLIBC_2.33 isgraph F
+GLIBC_2.33 isgraph_l F
+GLIBC_2.33 isinf F
+GLIBC_2.33 isinff F
+GLIBC_2.33 isinfl F
+GLIBC_2.33 islower F
+GLIBC_2.33 islower_l F
+GLIBC_2.33 isnan F
+GLIBC_2.33 isnanf F
+GLIBC_2.33 isnanl F
+GLIBC_2.33 isprint F
+GLIBC_2.33 isprint_l F
+GLIBC_2.33 ispunct F
+GLIBC_2.33 ispunct_l F
+GLIBC_2.33 isspace F
+GLIBC_2.33 isspace_l F
+GLIBC_2.33 isupper F
+GLIBC_2.33 isupper_l F
+GLIBC_2.33 iswalnum F
+GLIBC_2.33 iswalnum_l F
+GLIBC_2.33 iswalpha F
+GLIBC_2.33 iswalpha_l F
+GLIBC_2.33 iswblank F
+GLIBC_2.33 iswblank_l F
+GLIBC_2.33 iswcntrl F
+GLIBC_2.33 iswcntrl_l F
+GLIBC_2.33 iswctype F
+GLIBC_2.33 iswctype_l F
+GLIBC_2.33 iswdigit F
+GLIBC_2.33 iswdigit_l F
+GLIBC_2.33 iswgraph F
+GLIBC_2.33 iswgraph_l F
+GLIBC_2.33 iswlower F
+GLIBC_2.33 iswlower_l F
+GLIBC_2.33 iswprint F
+GLIBC_2.33 iswprint_l F
+GLIBC_2.33 iswpunct F
+GLIBC_2.33 iswpunct_l F
+GLIBC_2.33 iswspace F
+GLIBC_2.33 iswspace_l F
+GLIBC_2.33 iswupper F
+GLIBC_2.33 iswupper_l F
+GLIBC_2.33 iswxdigit F
+GLIBC_2.33 iswxdigit_l F
+GLIBC_2.33 isxdigit F
+GLIBC_2.33 isxdigit_l F
+GLIBC_2.33 jrand48 F
+GLIBC_2.33 jrand48_r F
+GLIBC_2.33 kill F
+GLIBC_2.33 killpg F
+GLIBC_2.33 klogctl F
+GLIBC_2.33 l64a F
+GLIBC_2.33 labs F
+GLIBC_2.33 lchmod F
+GLIBC_2.33 lchown F
+GLIBC_2.33 lckpwdf F
+GLIBC_2.33 lcong48 F
+GLIBC_2.33 lcong48_r F
+GLIBC_2.33 ldexp F
+GLIBC_2.33 ldexpf F
+GLIBC_2.33 ldexpl F
+GLIBC_2.33 ldiv F
+GLIBC_2.33 lfind F
+GLIBC_2.33 lgetxattr F
+GLIBC_2.33 link F
+GLIBC_2.33 linkat F
+GLIBC_2.33 listen F
+GLIBC_2.33 listxattr F
+GLIBC_2.33 llabs F
+GLIBC_2.33 lldiv F
+GLIBC_2.33 llistxattr F
+GLIBC_2.33 localeconv F
+GLIBC_2.33 localtime F
+GLIBC_2.33 localtime_r F
+GLIBC_2.33 lockf F
+GLIBC_2.33 lockf64 F
+GLIBC_2.33 longjmp F
+GLIBC_2.33 lrand48 F
+GLIBC_2.33 lrand48_r F
+GLIBC_2.33 lremovexattr F
+GLIBC_2.33 lsearch F
+GLIBC_2.33 lseek F
+GLIBC_2.33 lseek64 F
+GLIBC_2.33 lsetxattr F
+GLIBC_2.33 lutimes F
+GLIBC_2.33 madvise F
+GLIBC_2.33 makecontext F
+GLIBC_2.33 mallinfo F
+GLIBC_2.33 malloc F
+GLIBC_2.33 malloc_info F
+GLIBC_2.33 malloc_stats F
+GLIBC_2.33 malloc_trim F
+GLIBC_2.33 malloc_usable_size F
+GLIBC_2.33 mallopt F
+GLIBC_2.33 mallwatch D 0x4
+GLIBC_2.33 mblen F
+GLIBC_2.33 mbrlen F
+GLIBC_2.33 mbrtoc16 F
+GLIBC_2.33 mbrtoc32 F
+GLIBC_2.33 mbrtowc F
+GLIBC_2.33 mbsinit F
+GLIBC_2.33 mbsnrtowcs F
+GLIBC_2.33 mbsrtowcs F
+GLIBC_2.33 mbstowcs F
+GLIBC_2.33 mbtowc F
+GLIBC_2.33 mcheck F
+GLIBC_2.33 mcheck_check_all F
+GLIBC_2.33 mcheck_pedantic F
+GLIBC_2.33 memalign F
+GLIBC_2.33 memccpy F
+GLIBC_2.33 memchr F
+GLIBC_2.33 memcmp F
+GLIBC_2.33 memcpy F
+GLIBC_2.33 memfd_create F
+GLIBC_2.33 memfrob F
+GLIBC_2.33 memmem F
+GLIBC_2.33 memmove F
+GLIBC_2.33 mempcpy F
+GLIBC_2.33 memrchr F
+GLIBC_2.33 memset F
+GLIBC_2.33 mincore F
+GLIBC_2.33 mkdir F
+GLIBC_2.33 mkdirat F
+GLIBC_2.33 mkdtemp F
+GLIBC_2.33 mkfifo F
+GLIBC_2.33 mkfifoat F
+GLIBC_2.33 mkostemp F
+GLIBC_2.33 mkostemp64 F
+GLIBC_2.33 mkostemps F
+GLIBC_2.33 mkostemps64 F
+GLIBC_2.33 mkstemp F
+GLIBC_2.33 mkstemp64 F
+GLIBC_2.33 mkstemps F
+GLIBC_2.33 mkstemps64 F
+GLIBC_2.33 mktemp F
+GLIBC_2.33 mktime F
+GLIBC_2.33 mlock F
+GLIBC_2.33 mlock2 F
+GLIBC_2.33 mlockall F
+GLIBC_2.33 mmap F
+GLIBC_2.33 mmap64 F
+GLIBC_2.33 modf F
+GLIBC_2.33 modff F
+GLIBC_2.33 modfl F
+GLIBC_2.33 moncontrol F
+GLIBC_2.33 monstartup F
+GLIBC_2.33 mount F
+GLIBC_2.33 mprobe F
+GLIBC_2.33 mprotect F
+GLIBC_2.33 mrand48 F
+GLIBC_2.33 mrand48_r F
+GLIBC_2.33 mremap F
+GLIBC_2.33 msgctl F
+GLIBC_2.33 msgget F
+GLIBC_2.33 msgrcv F
+GLIBC_2.33 msgsnd F
+GLIBC_2.33 msync F
+GLIBC_2.33 mtrace F
+GLIBC_2.33 munlock F
+GLIBC_2.33 munlockall F
+GLIBC_2.33 munmap F
+GLIBC_2.33 muntrace F
+GLIBC_2.33 name_to_handle_at F
+GLIBC_2.33 nanosleep F
+GLIBC_2.33 newlocale F
+GLIBC_2.33 nftw F
+GLIBC_2.33 nftw64 F
+GLIBC_2.33 ngettext F
+GLIBC_2.33 nice F
+GLIBC_2.33 nl_langinfo F
+GLIBC_2.33 nl_langinfo_l F
+GLIBC_2.33 nrand48 F
+GLIBC_2.33 nrand48_r F
+GLIBC_2.33 ntohl F
+GLIBC_2.33 ntohs F
+GLIBC_2.33 ntp_adjtime F
+GLIBC_2.33 ntp_gettime F
+GLIBC_2.33 ntp_gettimex F
+GLIBC_2.33 obstack_alloc_failed_handler D 0x4
+GLIBC_2.33 obstack_exit_failure D 0x4
+GLIBC_2.33 obstack_free F
+GLIBC_2.33 obstack_printf F
+GLIBC_2.33 obstack_vprintf F
+GLIBC_2.33 on_exit F
+GLIBC_2.33 open F
+GLIBC_2.33 open64 F
+GLIBC_2.33 open_by_handle_at F
+GLIBC_2.33 open_memstream F
+GLIBC_2.33 open_wmemstream F
+GLIBC_2.33 openat F
+GLIBC_2.33 openat64 F
+GLIBC_2.33 opendir F
+GLIBC_2.33 openlog F
+GLIBC_2.33 optarg D 0x4
+GLIBC_2.33 opterr D 0x4
+GLIBC_2.33 optind D 0x4
+GLIBC_2.33 optopt D 0x4
+GLIBC_2.33 parse_printf_format F
+GLIBC_2.33 pathconf F
+GLIBC_2.33 pause F
+GLIBC_2.33 pclose F
+GLIBC_2.33 perror F
+GLIBC_2.33 personality F
+GLIBC_2.33 pipe F
+GLIBC_2.33 pipe2 F
+GLIBC_2.33 pivot_root F
+GLIBC_2.33 pkey_alloc F
+GLIBC_2.33 pkey_free F
+GLIBC_2.33 pkey_get F
+GLIBC_2.33 pkey_mprotect F
+GLIBC_2.33 pkey_set F
+GLIBC_2.33 poll F
+GLIBC_2.33 popen F
+GLIBC_2.33 posix_fadvise F
+GLIBC_2.33 posix_fadvise64 F
+GLIBC_2.33 posix_fallocate F
+GLIBC_2.33 posix_fallocate64 F
+GLIBC_2.33 posix_madvise F
+GLIBC_2.33 posix_memalign F
+GLIBC_2.33 posix_openpt F
+GLIBC_2.33 posix_spawn F
+GLIBC_2.33 posix_spawn_file_actions_addchdir_np F
+GLIBC_2.33 posix_spawn_file_actions_addclose F
+GLIBC_2.33 posix_spawn_file_actions_adddup2 F
+GLIBC_2.33 posix_spawn_file_actions_addfchdir_np F
+GLIBC_2.33 posix_spawn_file_actions_addopen F
+GLIBC_2.33 posix_spawn_file_actions_destroy F
+GLIBC_2.33 posix_spawn_file_actions_init F
+GLIBC_2.33 posix_spawnattr_destroy F
+GLIBC_2.33 posix_spawnattr_getflags F
+GLIBC_2.33 posix_spawnattr_getpgroup F
+GLIBC_2.33 posix_spawnattr_getschedparam F
+GLIBC_2.33 posix_spawnattr_getschedpolicy F
+GLIBC_2.33 posix_spawnattr_getsigdefault F
+GLIBC_2.33 posix_spawnattr_getsigmask F
+GLIBC_2.33 posix_spawnattr_init F
+GLIBC_2.33 posix_spawnattr_setflags F
+GLIBC_2.33 posix_spawnattr_setpgroup F
+GLIBC_2.33 posix_spawnattr_setschedparam F
+GLIBC_2.33 posix_spawnattr_setschedpolicy F
+GLIBC_2.33 posix_spawnattr_setsigdefault F
+GLIBC_2.33 posix_spawnattr_setsigmask F
+GLIBC_2.33 posix_spawnp F
+GLIBC_2.33 ppoll F
+GLIBC_2.33 prctl F
+GLIBC_2.33 pread F
+GLIBC_2.33 pread64 F
+GLIBC_2.33 preadv F
+GLIBC_2.33 preadv2 F
+GLIBC_2.33 preadv64 F
+GLIBC_2.33 preadv64v2 F
+GLIBC_2.33 printf F
+GLIBC_2.33 printf_size F
+GLIBC_2.33 printf_size_info F
+GLIBC_2.33 prlimit F
+GLIBC_2.33 prlimit64 F
+GLIBC_2.33 process_vm_readv F
+GLIBC_2.33 process_vm_writev F
+GLIBC_2.33 profil F
+GLIBC_2.33 program_invocation_name D 0x4
+GLIBC_2.33 program_invocation_short_name D 0x4
+GLIBC_2.33 pselect F
+GLIBC_2.33 psiginfo F
+GLIBC_2.33 psignal F
+GLIBC_2.33 pthread_attr_destroy F
+GLIBC_2.33 pthread_attr_getdetachstate F
+GLIBC_2.33 pthread_attr_getinheritsched F
+GLIBC_2.33 pthread_attr_getschedparam F
+GLIBC_2.33 pthread_attr_getschedpolicy F
+GLIBC_2.33 pthread_attr_getscope F
+GLIBC_2.33 pthread_attr_getsigmask_np F
+GLIBC_2.33 pthread_attr_init F
+GLIBC_2.33 pthread_attr_setaffinity_np F
+GLIBC_2.33 pthread_attr_setdetachstate F
+GLIBC_2.33 pthread_attr_setinheritsched F
+GLIBC_2.33 pthread_attr_setschedparam F
+GLIBC_2.33 pthread_attr_setschedpolicy F
+GLIBC_2.33 pthread_attr_setscope F
+GLIBC_2.33 pthread_attr_setsigmask_np F
+GLIBC_2.33 pthread_cond_broadcast F
+GLIBC_2.33 pthread_cond_destroy F
+GLIBC_2.33 pthread_cond_init F
+GLIBC_2.33 pthread_cond_signal F
+GLIBC_2.33 pthread_cond_timedwait F
+GLIBC_2.33 pthread_cond_wait F
+GLIBC_2.33 pthread_condattr_destroy F
+GLIBC_2.33 pthread_condattr_init F
+GLIBC_2.33 pthread_equal F
+GLIBC_2.33 pthread_exit F
+GLIBC_2.33 pthread_getaffinity_np F
+GLIBC_2.33 pthread_getattr_np F
+GLIBC_2.33 pthread_getschedparam F
+GLIBC_2.33 pthread_mutex_destroy F
+GLIBC_2.33 pthread_mutex_init F
+GLIBC_2.33 pthread_mutex_lock F
+GLIBC_2.33 pthread_mutex_unlock F
+GLIBC_2.33 pthread_self F
+GLIBC_2.33 pthread_setcancelstate F
+GLIBC_2.33 pthread_setcanceltype F
+GLIBC_2.33 pthread_setschedparam F
+GLIBC_2.33 pthread_sigmask F
+GLIBC_2.33 ptrace F
+GLIBC_2.33 ptsname F
+GLIBC_2.33 ptsname_r F
+GLIBC_2.33 putc F
+GLIBC_2.33 putc_unlocked F
+GLIBC_2.33 putchar F
+GLIBC_2.33 putchar_unlocked F
+GLIBC_2.33 putenv F
+GLIBC_2.33 putgrent F
+GLIBC_2.33 putpwent F
+GLIBC_2.33 puts F
+GLIBC_2.33 putsgent F
+GLIBC_2.33 putspent F
+GLIBC_2.33 pututline F
+GLIBC_2.33 pututxline F
+GLIBC_2.33 putw F
+GLIBC_2.33 putwc F
+GLIBC_2.33 putwc_unlocked F
+GLIBC_2.33 putwchar F
+GLIBC_2.33 putwchar_unlocked F
+GLIBC_2.33 pvalloc F
+GLIBC_2.33 pwrite F
+GLIBC_2.33 pwrite64 F
+GLIBC_2.33 pwritev F
+GLIBC_2.33 pwritev2 F
+GLIBC_2.33 pwritev64 F
+GLIBC_2.33 pwritev64v2 F
+GLIBC_2.33 qecvt F
+GLIBC_2.33 qecvt_r F
+GLIBC_2.33 qfcvt F
+GLIBC_2.33 qfcvt_r F
+GLIBC_2.33 qgcvt F
+GLIBC_2.33 qsort F
+GLIBC_2.33 qsort_r F
+GLIBC_2.33 quick_exit F
+GLIBC_2.33 quotactl F
+GLIBC_2.33 raise F
+GLIBC_2.33 rand F
+GLIBC_2.33 rand_r F
+GLIBC_2.33 random F
+GLIBC_2.33 random_r F
+GLIBC_2.33 rawmemchr F
+GLIBC_2.33 rcmd F
+GLIBC_2.33 rcmd_af F
+GLIBC_2.33 re_comp F
+GLIBC_2.33 re_compile_fastmap F
+GLIBC_2.33 re_compile_pattern F
+GLIBC_2.33 re_exec F
+GLIBC_2.33 re_match F
+GLIBC_2.33 re_match_2 F
+GLIBC_2.33 re_search F
+GLIBC_2.33 re_search_2 F
+GLIBC_2.33 re_set_registers F
+GLIBC_2.33 re_set_syntax F
+GLIBC_2.33 re_syntax_options D 0x4
+GLIBC_2.33 read F
+GLIBC_2.33 readahead F
+GLIBC_2.33 readdir F
+GLIBC_2.33 readdir64 F
+GLIBC_2.33 readdir64_r F
+GLIBC_2.33 readdir_r F
+GLIBC_2.33 readlink F
+GLIBC_2.33 readlinkat F
+GLIBC_2.33 readv F
+GLIBC_2.33 realloc F
+GLIBC_2.33 reallocarray F
+GLIBC_2.33 realpath F
+GLIBC_2.33 reboot F
+GLIBC_2.33 recv F
+GLIBC_2.33 recvfrom F
+GLIBC_2.33 recvmmsg F
+GLIBC_2.33 recvmsg F
+GLIBC_2.33 regcomp F
+GLIBC_2.33 regerror F
+GLIBC_2.33 regexec F
+GLIBC_2.33 regfree F
+GLIBC_2.33 register_printf_function F
+GLIBC_2.33 register_printf_modifier F
+GLIBC_2.33 register_printf_specifier F
+GLIBC_2.33 register_printf_type F
+GLIBC_2.33 remap_file_pages F
+GLIBC_2.33 remove F
+GLIBC_2.33 removexattr F
+GLIBC_2.33 remque F
+GLIBC_2.33 rename F
+GLIBC_2.33 renameat F
+GLIBC_2.33 renameat2 F
+GLIBC_2.33 revoke F
+GLIBC_2.33 rewind F
+GLIBC_2.33 rewinddir F
+GLIBC_2.33 rexec F
+GLIBC_2.33 rexec_af F
+GLIBC_2.33 rexecoptions D 0x4
+GLIBC_2.33 rindex F
+GLIBC_2.33 rmdir F
+GLIBC_2.33 rpmatch F
+GLIBC_2.33 rresvport F
+GLIBC_2.33 rresvport_af F
+GLIBC_2.33 ruserok F
+GLIBC_2.33 ruserok_af F
+GLIBC_2.33 ruserpass F
+GLIBC_2.33 sbrk F
+GLIBC_2.33 scalbn F
+GLIBC_2.33 scalbnf F
+GLIBC_2.33 scalbnl F
+GLIBC_2.33 scandir F
+GLIBC_2.33 scandir64 F
+GLIBC_2.33 scandirat F
+GLIBC_2.33 scandirat64 F
+GLIBC_2.33 scanf F
+GLIBC_2.33 sched_get_priority_max F
+GLIBC_2.33 sched_get_priority_min F
+GLIBC_2.33 sched_getaffinity F
+GLIBC_2.33 sched_getcpu F
+GLIBC_2.33 sched_getparam F
+GLIBC_2.33 sched_getscheduler F
+GLIBC_2.33 sched_rr_get_interval F
+GLIBC_2.33 sched_setaffinity F
+GLIBC_2.33 sched_setparam F
+GLIBC_2.33 sched_setscheduler F
+GLIBC_2.33 sched_yield F
+GLIBC_2.33 secure_getenv F
+GLIBC_2.33 seed48 F
+GLIBC_2.33 seed48_r F
+GLIBC_2.33 seekdir F
+GLIBC_2.33 select F
+GLIBC_2.33 semctl F
+GLIBC_2.33 semget F
+GLIBC_2.33 semop F
+GLIBC_2.33 semtimedop F
+GLIBC_2.33 send F
+GLIBC_2.33 sendfile F
+GLIBC_2.33 sendfile64 F
+GLIBC_2.33 sendmmsg F
+GLIBC_2.33 sendmsg F
+GLIBC_2.33 sendto F
+GLIBC_2.33 setaliasent F
+GLIBC_2.33 setbuf F
+GLIBC_2.33 setbuffer F
+GLIBC_2.33 setcontext F
+GLIBC_2.33 setdomainname F
+GLIBC_2.33 setegid F
+GLIBC_2.33 setenv F
+GLIBC_2.33 seteuid F
+GLIBC_2.33 setfsent F
+GLIBC_2.33 setfsgid F
+GLIBC_2.33 setfsuid F
+GLIBC_2.33 setgid F
+GLIBC_2.33 setgrent F
+GLIBC_2.33 setgroups F
+GLIBC_2.33 sethostent F
+GLIBC_2.33 sethostid F
+GLIBC_2.33 sethostname F
+GLIBC_2.33 setipv4sourcefilter F
+GLIBC_2.33 setitimer F
+GLIBC_2.33 setjmp F
+GLIBC_2.33 setlinebuf F
+GLIBC_2.33 setlocale F
+GLIBC_2.33 setlogin F
+GLIBC_2.33 setlogmask F
+GLIBC_2.33 setmntent F
+GLIBC_2.33 setnetent F
+GLIBC_2.33 setnetgrent F
+GLIBC_2.33 setns F
+GLIBC_2.33 setpgid F
+GLIBC_2.33 setpgrp F
+GLIBC_2.33 setpriority F
+GLIBC_2.33 setprotoent F
+GLIBC_2.33 setpwent F
+GLIBC_2.33 setregid F
+GLIBC_2.33 setresgid F
+GLIBC_2.33 setresuid F
+GLIBC_2.33 setreuid F
+GLIBC_2.33 setrlimit F
+GLIBC_2.33 setrlimit64 F
+GLIBC_2.33 setrpcent F
+GLIBC_2.33 setservent F
+GLIBC_2.33 setsgent F
+GLIBC_2.33 setsid F
+GLIBC_2.33 setsockopt F
+GLIBC_2.33 setsourcefilter F
+GLIBC_2.33 setspent F
+GLIBC_2.33 setstate F
+GLIBC_2.33 setstate_r F
+GLIBC_2.33 settimeofday F
+GLIBC_2.33 setttyent F
+GLIBC_2.33 setuid F
+GLIBC_2.33 setusershell F
+GLIBC_2.33 setutent F
+GLIBC_2.33 setutxent F
+GLIBC_2.33 setvbuf F
+GLIBC_2.33 setxattr F
+GLIBC_2.33 sgetsgent F
+GLIBC_2.33 sgetsgent_r F
+GLIBC_2.33 sgetspent F
+GLIBC_2.33 sgetspent_r F
+GLIBC_2.33 shmat F
+GLIBC_2.33 shmctl F
+GLIBC_2.33 shmdt F
+GLIBC_2.33 shmget F
+GLIBC_2.33 shutdown F
+GLIBC_2.33 sigabbrev_np F
+GLIBC_2.33 sigaction F
+GLIBC_2.33 sigaddset F
+GLIBC_2.33 sigaltstack F
+GLIBC_2.33 sigandset F
+GLIBC_2.33 sigblock F
+GLIBC_2.33 sigdelset F
+GLIBC_2.33 sigdescr_np F
+GLIBC_2.33 sigemptyset F
+GLIBC_2.33 sigfillset F
+GLIBC_2.33 siggetmask F
+GLIBC_2.33 sighold F
+GLIBC_2.33 sigignore F
+GLIBC_2.33 siginterrupt F
+GLIBC_2.33 sigisemptyset F
+GLIBC_2.33 sigismember F
+GLIBC_2.33 siglongjmp F
+GLIBC_2.33 signal F
+GLIBC_2.33 signalfd F
+GLIBC_2.33 sigorset F
+GLIBC_2.33 sigpause F
+GLIBC_2.33 sigpending F
+GLIBC_2.33 sigprocmask F
+GLIBC_2.33 sigqueue F
+GLIBC_2.33 sigrelse F
+GLIBC_2.33 sigreturn F
+GLIBC_2.33 sigset F
+GLIBC_2.33 sigsetmask F
+GLIBC_2.33 sigstack F
+GLIBC_2.33 sigsuspend F
+GLIBC_2.33 sigtimedwait F
+GLIBC_2.33 sigwait F
+GLIBC_2.33 sigwaitinfo F
+GLIBC_2.33 sleep F
+GLIBC_2.33 snprintf F
+GLIBC_2.33 sockatmark F
+GLIBC_2.33 socket F
+GLIBC_2.33 socketpair F
+GLIBC_2.33 splice F
+GLIBC_2.33 sprintf F
+GLIBC_2.33 sprofil F
+GLIBC_2.33 srand F
+GLIBC_2.33 srand48 F
+GLIBC_2.33 srand48_r F
+GLIBC_2.33 srandom F
+GLIBC_2.33 srandom_r F
+GLIBC_2.33 sscanf F
+GLIBC_2.33 ssignal F
+GLIBC_2.33 statfs F
+GLIBC_2.33 statfs64 F
+GLIBC_2.33 statvfs F
+GLIBC_2.33 statvfs64 F
+GLIBC_2.33 statx F
+GLIBC_2.33 stderr D 0x4
+GLIBC_2.33 stdin D 0x4
+GLIBC_2.33 stdout D 0x4
+GLIBC_2.33 stpcpy F
+GLIBC_2.33 stpncpy F
+GLIBC_2.33 strcasecmp F
+GLIBC_2.33 strcasecmp_l F
+GLIBC_2.33 strcasestr F
+GLIBC_2.33 strcat F
+GLIBC_2.33 strchr F
+GLIBC_2.33 strchrnul F
+GLIBC_2.33 strcmp F
+GLIBC_2.33 strcoll F
+GLIBC_2.33 strcoll_l F
+GLIBC_2.33 strcpy F
+GLIBC_2.33 strcspn F
+GLIBC_2.33 strdup F
+GLIBC_2.33 strerror F
+GLIBC_2.33 strerror_l F
+GLIBC_2.33 strerror_r F
+GLIBC_2.33 strerrordesc_np F
+GLIBC_2.33 strerrorname_np F
+GLIBC_2.33 strfmon F
+GLIBC_2.33 strfmon_l F
+GLIBC_2.33 strfromd F
+GLIBC_2.33 strfromf F
+GLIBC_2.33 strfromf128 F
+GLIBC_2.33 strfromf32 F
+GLIBC_2.33 strfromf32x F
+GLIBC_2.33 strfromf64 F
+GLIBC_2.33 strfromf64x F
+GLIBC_2.33 strfroml F
+GLIBC_2.33 strfry F
+GLIBC_2.33 strftime F
+GLIBC_2.33 strftime_l F
+GLIBC_2.33 strlen F
+GLIBC_2.33 strncasecmp F
+GLIBC_2.33 strncasecmp_l F
+GLIBC_2.33 strncat F
+GLIBC_2.33 strncmp F
+GLIBC_2.33 strncpy F
+GLIBC_2.33 strndup F
+GLIBC_2.33 strnlen F
+GLIBC_2.33 strpbrk F
+GLIBC_2.33 strptime F
+GLIBC_2.33 strptime_l F
+GLIBC_2.33 strrchr F
+GLIBC_2.33 strsep F
+GLIBC_2.33 strsignal F
+GLIBC_2.33 strspn F
+GLIBC_2.33 strstr F
+GLIBC_2.33 strtod F
+GLIBC_2.33 strtod_l F
+GLIBC_2.33 strtof F
+GLIBC_2.33 strtof128 F
+GLIBC_2.33 strtof128_l F
+GLIBC_2.33 strtof32 F
+GLIBC_2.33 strtof32_l F
+GLIBC_2.33 strtof32x F
+GLIBC_2.33 strtof32x_l F
+GLIBC_2.33 strtof64 F
+GLIBC_2.33 strtof64_l F
+GLIBC_2.33 strtof64x F
+GLIBC_2.33 strtof64x_l F
+GLIBC_2.33 strtof_l F
+GLIBC_2.33 strtoimax F
+GLIBC_2.33 strtok F
+GLIBC_2.33 strtok_r F
+GLIBC_2.33 strtol F
+GLIBC_2.33 strtol_l F
+GLIBC_2.33 strtold F
+GLIBC_2.33 strtold_l F
+GLIBC_2.33 strtoll F
+GLIBC_2.33 strtoll_l F
+GLIBC_2.33 strtoq F
+GLIBC_2.33 strtoul F
+GLIBC_2.33 strtoul_l F
+GLIBC_2.33 strtoull F
+GLIBC_2.33 strtoull_l F
+GLIBC_2.33 strtoumax F
+GLIBC_2.33 strtouq F
+GLIBC_2.33 strverscmp F
+GLIBC_2.33 strxfrm F
+GLIBC_2.33 strxfrm_l F
+GLIBC_2.33 stty F
+GLIBC_2.33 swab F
+GLIBC_2.33 swapcontext F
+GLIBC_2.33 swapoff F
+GLIBC_2.33 swapon F
+GLIBC_2.33 swprintf F
+GLIBC_2.33 swscanf F
+GLIBC_2.33 symlink F
+GLIBC_2.33 symlinkat F
+GLIBC_2.33 sync F
+GLIBC_2.33 sync_file_range F
+GLIBC_2.33 syncfs F
+GLIBC_2.33 syscall F
+GLIBC_2.33 sysconf F
+GLIBC_2.33 sysinfo F
+GLIBC_2.33 syslog F
+GLIBC_2.33 system F
+GLIBC_2.33 sysv_signal F
+GLIBC_2.33 tcdrain F
+GLIBC_2.33 tcflow F
+GLIBC_2.33 tcflush F
+GLIBC_2.33 tcgetattr F
+GLIBC_2.33 tcgetpgrp F
+GLIBC_2.33 tcgetsid F
+GLIBC_2.33 tcsendbreak F
+GLIBC_2.33 tcsetattr F
+GLIBC_2.33 tcsetpgrp F
+GLIBC_2.33 tdelete F
+GLIBC_2.33 tdestroy F
+GLIBC_2.33 tee F
+GLIBC_2.33 telldir F
+GLIBC_2.33 tempnam F
+GLIBC_2.33 textdomain F
+GLIBC_2.33 tfind F
+GLIBC_2.33 tgkill F
+GLIBC_2.33 thrd_current F
+GLIBC_2.33 thrd_equal F
+GLIBC_2.33 thrd_sleep F
+GLIBC_2.33 thrd_yield F
+GLIBC_2.33 time F
+GLIBC_2.33 timegm F
+GLIBC_2.33 timelocal F
+GLIBC_2.33 timerfd_create F
+GLIBC_2.33 timerfd_gettime F
+GLIBC_2.33 timerfd_settime F
+GLIBC_2.33 times F
+GLIBC_2.33 timespec_get F
+GLIBC_2.33 timezone D 0x4
+GLIBC_2.33 tmpfile F
+GLIBC_2.33 tmpfile64 F
+GLIBC_2.33 tmpnam F
+GLIBC_2.33 tmpnam_r F
+GLIBC_2.33 toascii F
+GLIBC_2.33 tolower F
+GLIBC_2.33 tolower_l F
+GLIBC_2.33 toupper F
+GLIBC_2.33 toupper_l F
+GLIBC_2.33 towctrans F
+GLIBC_2.33 towctrans_l F
+GLIBC_2.33 towlower F
+GLIBC_2.33 towlower_l F
+GLIBC_2.33 towupper F
+GLIBC_2.33 towupper_l F
+GLIBC_2.33 tr_break F
+GLIBC_2.33 truncate F
+GLIBC_2.33 truncate64 F
+GLIBC_2.33 tsearch F
+GLIBC_2.33 ttyname F
+GLIBC_2.33 ttyname_r F
+GLIBC_2.33 ttyslot F
+GLIBC_2.33 twalk F
+GLIBC_2.33 twalk_r F
+GLIBC_2.33 tzname D 0x8
+GLIBC_2.33 tzset F
+GLIBC_2.33 ualarm F
+GLIBC_2.33 ulckpwdf F
+GLIBC_2.33 ulimit F
+GLIBC_2.33 umask F
+GLIBC_2.33 umount F
+GLIBC_2.33 umount2 F
+GLIBC_2.33 uname F
+GLIBC_2.33 ungetc F
+GLIBC_2.33 ungetwc F
+GLIBC_2.33 unlink F
+GLIBC_2.33 unlinkat F
+GLIBC_2.33 unlockpt F
+GLIBC_2.33 unsetenv F
+GLIBC_2.33 unshare F
+GLIBC_2.33 updwtmp F
+GLIBC_2.33 updwtmpx F
+GLIBC_2.33 uselocale F
+GLIBC_2.33 usleep F
+GLIBC_2.33 utime F
+GLIBC_2.33 utimensat F
+GLIBC_2.33 utimes F
+GLIBC_2.33 utmpname F
+GLIBC_2.33 utmpxname F
+GLIBC_2.33 valloc F
+GLIBC_2.33 vasprintf F
+GLIBC_2.33 vdprintf F
+GLIBC_2.33 verr F
+GLIBC_2.33 verrx F
+GLIBC_2.33 versionsort F
+GLIBC_2.33 versionsort64 F
+GLIBC_2.33 vfork F
+GLIBC_2.33 vfprintf F
+GLIBC_2.33 vfscanf F
+GLIBC_2.33 vfwprintf F
+GLIBC_2.33 vfwscanf F
+GLIBC_2.33 vhangup F
+GLIBC_2.33 vlimit F
+GLIBC_2.33 vmsplice F
+GLIBC_2.33 vprintf F
+GLIBC_2.33 vscanf F
+GLIBC_2.33 vsnprintf F
+GLIBC_2.33 vsprintf F
+GLIBC_2.33 vsscanf F
+GLIBC_2.33 vswprintf F
+GLIBC_2.33 vswscanf F
+GLIBC_2.33 vsyslog F
+GLIBC_2.33 vtimes F
+GLIBC_2.33 vwarn F
+GLIBC_2.33 vwarnx F
+GLIBC_2.33 vwprintf F
+GLIBC_2.33 vwscanf F
+GLIBC_2.33 wait F
+GLIBC_2.33 wait3 F
+GLIBC_2.33 wait4 F
+GLIBC_2.33 waitid F
+GLIBC_2.33 waitpid F
+GLIBC_2.33 warn F
+GLIBC_2.33 warnx F
+GLIBC_2.33 wcpcpy F
+GLIBC_2.33 wcpncpy F
+GLIBC_2.33 wcrtomb F
+GLIBC_2.33 wcscasecmp F
+GLIBC_2.33 wcscasecmp_l F
+GLIBC_2.33 wcscat F
+GLIBC_2.33 wcschr F
+GLIBC_2.33 wcschrnul F
+GLIBC_2.33 wcscmp F
+GLIBC_2.33 wcscoll F
+GLIBC_2.33 wcscoll_l F
+GLIBC_2.33 wcscpy F
+GLIBC_2.33 wcscspn F
+GLIBC_2.33 wcsdup F
+GLIBC_2.33 wcsftime F
+GLIBC_2.33 wcsftime_l F
+GLIBC_2.33 wcslen F
+GLIBC_2.33 wcsncasecmp F
+GLIBC_2.33 wcsncasecmp_l F
+GLIBC_2.33 wcsncat F
+GLIBC_2.33 wcsncmp F
+GLIBC_2.33 wcsncpy F
+GLIBC_2.33 wcsnlen F
+GLIBC_2.33 wcsnrtombs F
+GLIBC_2.33 wcspbrk F
+GLIBC_2.33 wcsrchr F
+GLIBC_2.33 wcsrtombs F
+GLIBC_2.33 wcsspn F
+GLIBC_2.33 wcsstr F
+GLIBC_2.33 wcstod F
+GLIBC_2.33 wcstod_l F
+GLIBC_2.33 wcstof F
+GLIBC_2.33 wcstof128 F
+GLIBC_2.33 wcstof128_l F
+GLIBC_2.33 wcstof32 F
+GLIBC_2.33 wcstof32_l F
+GLIBC_2.33 wcstof32x F
+GLIBC_2.33 wcstof32x_l F
+GLIBC_2.33 wcstof64 F
+GLIBC_2.33 wcstof64_l F
+GLIBC_2.33 wcstof64x F
+GLIBC_2.33 wcstof64x_l F
+GLIBC_2.33 wcstof_l F
+GLIBC_2.33 wcstoimax F
+GLIBC_2.33 wcstok F
+GLIBC_2.33 wcstol F
+GLIBC_2.33 wcstol_l F
+GLIBC_2.33 wcstold F
+GLIBC_2.33 wcstold_l F
+GLIBC_2.33 wcstoll F
+GLIBC_2.33 wcstoll_l F
+GLIBC_2.33 wcstombs F
+GLIBC_2.33 wcstoq F
+GLIBC_2.33 wcstoul F
+GLIBC_2.33 wcstoul_l F
+GLIBC_2.33 wcstoull F
+GLIBC_2.33 wcstoull_l F
+GLIBC_2.33 wcstoumax F
+GLIBC_2.33 wcstouq F
+GLIBC_2.33 wcswcs F
+GLIBC_2.33 wcswidth F
+GLIBC_2.33 wcsxfrm F
+GLIBC_2.33 wcsxfrm_l F
+GLIBC_2.33 wctob F
+GLIBC_2.33 wctomb F
+GLIBC_2.33 wctrans F
+GLIBC_2.33 wctrans_l F
+GLIBC_2.33 wctype F
+GLIBC_2.33 wctype_l F
+GLIBC_2.33 wcwidth F
+GLIBC_2.33 wmemchr F
+GLIBC_2.33 wmemcmp F
+GLIBC_2.33 wmemcpy F
+GLIBC_2.33 wmemmove F
+GLIBC_2.33 wmempcpy F
+GLIBC_2.33 wmemset F
+GLIBC_2.33 wordexp F
+GLIBC_2.33 wordfree F
+GLIBC_2.33 wprintf F
+GLIBC_2.33 write F
+GLIBC_2.33 writev F
+GLIBC_2.33 wscanf F
diff --git a/sysdeps/unix/sysv/linux/riscv/rv32/libcrypt.abilist b/sysdeps/unix/sysv/linux/riscv/rv32/libcrypt.abilist
new file mode 100644
index 0000000000..f6ed52d39f
--- /dev/null
+++ b/sysdeps/unix/sysv/linux/riscv/rv32/libcrypt.abilist
@@ -0,0 +1,2 @@
+GLIBC_2.33 crypt F
+GLIBC_2.33 crypt_r F
diff --git a/sysdeps/unix/sysv/linux/riscv/rv32/libdl.abilist b/sysdeps/unix/sysv/linux/riscv/rv32/libdl.abilist
new file mode 100644
index 0000000000..ac5d2d0e9e
--- /dev/null
+++ b/sysdeps/unix/sysv/linux/riscv/rv32/libdl.abilist
@@ -0,0 +1,9 @@
+GLIBC_2.33 dladdr F
+GLIBC_2.33 dladdr1 F
+GLIBC_2.33 dlclose F
+GLIBC_2.33 dlerror F
+GLIBC_2.33 dlinfo F
+GLIBC_2.33 dlmopen F
+GLIBC_2.33 dlopen F
+GLIBC_2.33 dlsym F
+GLIBC_2.33 dlvsym F
diff --git a/sysdeps/unix/sysv/linux/riscv/rv32/libm.abilist b/sysdeps/unix/sysv/linux/riscv/rv32/libm.abilist
new file mode 100644
index 0000000000..c701b07a9a
--- /dev/null
+++ b/sysdeps/unix/sysv/linux/riscv/rv32/libm.abilist
@@ -0,0 +1,940 @@
+GLIBC_2.33 __clog10 F
+GLIBC_2.33 __clog10f F
+GLIBC_2.33 __clog10l F
+GLIBC_2.33 __finite F
+GLIBC_2.33 __finitef F
+GLIBC_2.33 __finitel F
+GLIBC_2.33 __fpclassify F
+GLIBC_2.33 __fpclassifyf F
+GLIBC_2.33 __fpclassifyl F
+GLIBC_2.33 __iseqsig F
+GLIBC_2.33 __iseqsigf F
+GLIBC_2.33 __iseqsigl F
+GLIBC_2.33 __issignaling F
+GLIBC_2.33 __issignalingf F
+GLIBC_2.33 __issignalingl F
+GLIBC_2.33 __signbit F
+GLIBC_2.33 __signbitf F
+GLIBC_2.33 __signbitl F
+GLIBC_2.33 __signgam D 0x4
+GLIBC_2.33 acos F
+GLIBC_2.33 acosf F
+GLIBC_2.33 acosf128 F
+GLIBC_2.33 acosf32 F
+GLIBC_2.33 acosf32x F
+GLIBC_2.33 acosf64 F
+GLIBC_2.33 acosf64x F
+GLIBC_2.33 acosh F
+GLIBC_2.33 acoshf F
+GLIBC_2.33 acoshf128 F
+GLIBC_2.33 acoshf32 F
+GLIBC_2.33 acoshf32x F
+GLIBC_2.33 acoshf64 F
+GLIBC_2.33 acoshf64x F
+GLIBC_2.33 acoshl F
+GLIBC_2.33 acosl F
+GLIBC_2.33 asin F
+GLIBC_2.33 asinf F
+GLIBC_2.33 asinf128 F
+GLIBC_2.33 asinf32 F
+GLIBC_2.33 asinf32x F
+GLIBC_2.33 asinf64 F
+GLIBC_2.33 asinf64x F
+GLIBC_2.33 asinh F
+GLIBC_2.33 asinhf F
+GLIBC_2.33 asinhf128 F
+GLIBC_2.33 asinhf32 F
+GLIBC_2.33 asinhf32x F
+GLIBC_2.33 asinhf64 F
+GLIBC_2.33 asinhf64x F
+GLIBC_2.33 asinhl F
+GLIBC_2.33 asinl F
+GLIBC_2.33 atan F
+GLIBC_2.33 atan2 F
+GLIBC_2.33 atan2f F
+GLIBC_2.33 atan2f128 F
+GLIBC_2.33 atan2f32 F
+GLIBC_2.33 atan2f32x F
+GLIBC_2.33 atan2f64 F
+GLIBC_2.33 atan2f64x F
+GLIBC_2.33 atan2l F
+GLIBC_2.33 atanf F
+GLIBC_2.33 atanf128 F
+GLIBC_2.33 atanf32 F
+GLIBC_2.33 atanf32x F
+GLIBC_2.33 atanf64 F
+GLIBC_2.33 atanf64x F
+GLIBC_2.33 atanh F
+GLIBC_2.33 atanhf F
+GLIBC_2.33 atanhf128 F
+GLIBC_2.33 atanhf32 F
+GLIBC_2.33 atanhf32x F
+GLIBC_2.33 atanhf64 F
+GLIBC_2.33 atanhf64x F
+GLIBC_2.33 atanhl F
+GLIBC_2.33 atanl F
+GLIBC_2.33 cabs F
+GLIBC_2.33 cabsf F
+GLIBC_2.33 cabsf128 F
+GLIBC_2.33 cabsf32 F
+GLIBC_2.33 cabsf32x F
+GLIBC_2.33 cabsf64 F
+GLIBC_2.33 cabsf64x F
+GLIBC_2.33 cabsl F
+GLIBC_2.33 cacos F
+GLIBC_2.33 cacosf F
+GLIBC_2.33 cacosf128 F
+GLIBC_2.33 cacosf32 F
+GLIBC_2.33 cacosf32x F
+GLIBC_2.33 cacosf64 F
+GLIBC_2.33 cacosf64x F
+GLIBC_2.33 cacosh F
+GLIBC_2.33 cacoshf F
+GLIBC_2.33 cacoshf128 F
+GLIBC_2.33 cacoshf32 F
+GLIBC_2.33 cacoshf32x F
+GLIBC_2.33 cacoshf64 F
+GLIBC_2.33 cacoshf64x F
+GLIBC_2.33 cacoshl F
+GLIBC_2.33 cacosl F
+GLIBC_2.33 canonicalize F
+GLIBC_2.33 canonicalizef F
+GLIBC_2.33 canonicalizef128 F
+GLIBC_2.33 canonicalizef32 F
+GLIBC_2.33 canonicalizef32x F
+GLIBC_2.33 canonicalizef64 F
+GLIBC_2.33 canonicalizef64x F
+GLIBC_2.33 canonicalizel F
+GLIBC_2.33 carg F
+GLIBC_2.33 cargf F
+GLIBC_2.33 cargf128 F
+GLIBC_2.33 cargf32 F
+GLIBC_2.33 cargf32x F
+GLIBC_2.33 cargf64 F
+GLIBC_2.33 cargf64x F
+GLIBC_2.33 cargl F
+GLIBC_2.33 casin F
+GLIBC_2.33 casinf F
+GLIBC_2.33 casinf128 F
+GLIBC_2.33 casinf32 F
+GLIBC_2.33 casinf32x F
+GLIBC_2.33 casinf64 F
+GLIBC_2.33 casinf64x F
+GLIBC_2.33 casinh F
+GLIBC_2.33 casinhf F
+GLIBC_2.33 casinhf128 F
+GLIBC_2.33 casinhf32 F
+GLIBC_2.33 casinhf32x F
+GLIBC_2.33 casinhf64 F
+GLIBC_2.33 casinhf64x F
+GLIBC_2.33 casinhl F
+GLIBC_2.33 casinl F
+GLIBC_2.33 catan F
+GLIBC_2.33 catanf F
+GLIBC_2.33 catanf128 F
+GLIBC_2.33 catanf32 F
+GLIBC_2.33 catanf32x F
+GLIBC_2.33 catanf64 F
+GLIBC_2.33 catanf64x F
+GLIBC_2.33 catanh F
+GLIBC_2.33 catanhf F
+GLIBC_2.33 catanhf128 F
+GLIBC_2.33 catanhf32 F
+GLIBC_2.33 catanhf32x F
+GLIBC_2.33 catanhf64 F
+GLIBC_2.33 catanhf64x F
+GLIBC_2.33 catanhl F
+GLIBC_2.33 catanl F
+GLIBC_2.33 cbrt F
+GLIBC_2.33 cbrtf F
+GLIBC_2.33 cbrtf128 F
+GLIBC_2.33 cbrtf32 F
+GLIBC_2.33 cbrtf32x F
+GLIBC_2.33 cbrtf64 F
+GLIBC_2.33 cbrtf64x F
+GLIBC_2.33 cbrtl F
+GLIBC_2.33 ccos F
+GLIBC_2.33 ccosf F
+GLIBC_2.33 ccosf128 F
+GLIBC_2.33 ccosf32 F
+GLIBC_2.33 ccosf32x F
+GLIBC_2.33 ccosf64 F
+GLIBC_2.33 ccosf64x F
+GLIBC_2.33 ccosh F
+GLIBC_2.33 ccoshf F
+GLIBC_2.33 ccoshf128 F
+GLIBC_2.33 ccoshf32 F
+GLIBC_2.33 ccoshf32x F
+GLIBC_2.33 ccoshf64 F
+GLIBC_2.33 ccoshf64x F
+GLIBC_2.33 ccoshl F
+GLIBC_2.33 ccosl F
+GLIBC_2.33 ceil F
+GLIBC_2.33 ceilf F
+GLIBC_2.33 ceilf128 F
+GLIBC_2.33 ceilf32 F
+GLIBC_2.33 ceilf32x F
+GLIBC_2.33 ceilf64 F
+GLIBC_2.33 ceilf64x F
+GLIBC_2.33 ceill F
+GLIBC_2.33 cexp F
+GLIBC_2.33 cexpf F
+GLIBC_2.33 cexpf128 F
+GLIBC_2.33 cexpf32 F
+GLIBC_2.33 cexpf32x F
+GLIBC_2.33 cexpf64 F
+GLIBC_2.33 cexpf64x F
+GLIBC_2.33 cexpl F
+GLIBC_2.33 cimag F
+GLIBC_2.33 cimagf F
+GLIBC_2.33 cimagf128 F
+GLIBC_2.33 cimagf32 F
+GLIBC_2.33 cimagf32x F
+GLIBC_2.33 cimagf64 F
+GLIBC_2.33 cimagf64x F
+GLIBC_2.33 cimagl F
+GLIBC_2.33 clog F
+GLIBC_2.33 clog10 F
+GLIBC_2.33 clog10f F
+GLIBC_2.33 clog10f128 F
+GLIBC_2.33 clog10f32 F
+GLIBC_2.33 clog10f32x F
+GLIBC_2.33 clog10f64 F
+GLIBC_2.33 clog10f64x F
+GLIBC_2.33 clog10l F
+GLIBC_2.33 clogf F
+GLIBC_2.33 clogf128 F
+GLIBC_2.33 clogf32 F
+GLIBC_2.33 clogf32x F
+GLIBC_2.33 clogf64 F
+GLIBC_2.33 clogf64x F
+GLIBC_2.33 clogl F
+GLIBC_2.33 conj F
+GLIBC_2.33 conjf F
+GLIBC_2.33 conjf128 F
+GLIBC_2.33 conjf32 F
+GLIBC_2.33 conjf32x F
+GLIBC_2.33 conjf64 F
+GLIBC_2.33 conjf64x F
+GLIBC_2.33 conjl F
+GLIBC_2.33 copysign F
+GLIBC_2.33 copysignf F
+GLIBC_2.33 copysignf128 F
+GLIBC_2.33 copysignf32 F
+GLIBC_2.33 copysignf32x F
+GLIBC_2.33 copysignf64 F
+GLIBC_2.33 copysignf64x F
+GLIBC_2.33 copysignl F
+GLIBC_2.33 cos F
+GLIBC_2.33 cosf F
+GLIBC_2.33 cosf128 F
+GLIBC_2.33 cosf32 F
+GLIBC_2.33 cosf32x F
+GLIBC_2.33 cosf64 F
+GLIBC_2.33 cosf64x F
+GLIBC_2.33 cosh F
+GLIBC_2.33 coshf F
+GLIBC_2.33 coshf128 F
+GLIBC_2.33 coshf32 F
+GLIBC_2.33 coshf32x F
+GLIBC_2.33 coshf64 F
+GLIBC_2.33 coshf64x F
+GLIBC_2.33 coshl F
+GLIBC_2.33 cosl F
+GLIBC_2.33 cpow F
+GLIBC_2.33 cpowf F
+GLIBC_2.33 cpowf128 F
+GLIBC_2.33 cpowf32 F
+GLIBC_2.33 cpowf32x F
+GLIBC_2.33 cpowf64 F
+GLIBC_2.33 cpowf64x F
+GLIBC_2.33 cpowl F
+GLIBC_2.33 cproj F
+GLIBC_2.33 cprojf F
+GLIBC_2.33 cprojf128 F
+GLIBC_2.33 cprojf32 F
+GLIBC_2.33 cprojf32x F
+GLIBC_2.33 cprojf64 F
+GLIBC_2.33 cprojf64x F
+GLIBC_2.33 cprojl F
+GLIBC_2.33 creal F
+GLIBC_2.33 crealf F
+GLIBC_2.33 crealf128 F
+GLIBC_2.33 crealf32 F
+GLIBC_2.33 crealf32x F
+GLIBC_2.33 crealf64 F
+GLIBC_2.33 crealf64x F
+GLIBC_2.33 creall F
+GLIBC_2.33 csin F
+GLIBC_2.33 csinf F
+GLIBC_2.33 csinf128 F
+GLIBC_2.33 csinf32 F
+GLIBC_2.33 csinf32x F
+GLIBC_2.33 csinf64 F
+GLIBC_2.33 csinf64x F
+GLIBC_2.33 csinh F
+GLIBC_2.33 csinhf F
+GLIBC_2.33 csinhf128 F
+GLIBC_2.33 csinhf32 F
+GLIBC_2.33 csinhf32x F
+GLIBC_2.33 csinhf64 F
+GLIBC_2.33 csinhf64x F
+GLIBC_2.33 csinhl F
+GLIBC_2.33 csinl F
+GLIBC_2.33 csqrt F
+GLIBC_2.33 csqrtf F
+GLIBC_2.33 csqrtf128 F
+GLIBC_2.33 csqrtf32 F
+GLIBC_2.33 csqrtf32x F
+GLIBC_2.33 csqrtf64 F
+GLIBC_2.33 csqrtf64x F
+GLIBC_2.33 csqrtl F
+GLIBC_2.33 ctan F
+GLIBC_2.33 ctanf F
+GLIBC_2.33 ctanf128 F
+GLIBC_2.33 ctanf32 F
+GLIBC_2.33 ctanf32x F
+GLIBC_2.33 ctanf64 F
+GLIBC_2.33 ctanf64x F
+GLIBC_2.33 ctanh F
+GLIBC_2.33 ctanhf F
+GLIBC_2.33 ctanhf128 F
+GLIBC_2.33 ctanhf32 F
+GLIBC_2.33 ctanhf32x F
+GLIBC_2.33 ctanhf64 F
+GLIBC_2.33 ctanhf64x F
+GLIBC_2.33 ctanhl F
+GLIBC_2.33 ctanl F
+GLIBC_2.33 daddl F
+GLIBC_2.33 ddivl F
+GLIBC_2.33 dmull F
+GLIBC_2.33 drem F
+GLIBC_2.33 dremf F
+GLIBC_2.33 dreml F
+GLIBC_2.33 dsubl F
+GLIBC_2.33 erf F
+GLIBC_2.33 erfc F
+GLIBC_2.33 erfcf F
+GLIBC_2.33 erfcf128 F
+GLIBC_2.33 erfcf32 F
+GLIBC_2.33 erfcf32x F
+GLIBC_2.33 erfcf64 F
+GLIBC_2.33 erfcf64x F
+GLIBC_2.33 erfcl F
+GLIBC_2.33 erff F
+GLIBC_2.33 erff128 F
+GLIBC_2.33 erff32 F
+GLIBC_2.33 erff32x F
+GLIBC_2.33 erff64 F
+GLIBC_2.33 erff64x F
+GLIBC_2.33 erfl F
+GLIBC_2.33 exp F
+GLIBC_2.33 exp10 F
+GLIBC_2.33 exp10f F
+GLIBC_2.33 exp10f128 F
+GLIBC_2.33 exp10f32 F
+GLIBC_2.33 exp10f32x F
+GLIBC_2.33 exp10f64 F
+GLIBC_2.33 exp10f64x F
+GLIBC_2.33 exp10l F
+GLIBC_2.33 exp2 F
+GLIBC_2.33 exp2f F
+GLIBC_2.33 exp2f128 F
+GLIBC_2.33 exp2f32 F
+GLIBC_2.33 exp2f32x F
+GLIBC_2.33 exp2f64 F
+GLIBC_2.33 exp2f64x F
+GLIBC_2.33 exp2l F
+GLIBC_2.33 expf F
+GLIBC_2.33 expf128 F
+GLIBC_2.33 expf32 F
+GLIBC_2.33 expf32x F
+GLIBC_2.33 expf64 F
+GLIBC_2.33 expf64x F
+GLIBC_2.33 expl F
+GLIBC_2.33 expm1 F
+GLIBC_2.33 expm1f F
+GLIBC_2.33 expm1f128 F
+GLIBC_2.33 expm1f32 F
+GLIBC_2.33 expm1f32x F
+GLIBC_2.33 expm1f64 F
+GLIBC_2.33 expm1f64x F
+GLIBC_2.33 expm1l F
+GLIBC_2.33 f32addf128 F
+GLIBC_2.33 f32addf32x F
+GLIBC_2.33 f32addf64 F
+GLIBC_2.33 f32addf64x F
+GLIBC_2.33 f32divf128 F
+GLIBC_2.33 f32divf32x F
+GLIBC_2.33 f32divf64 F
+GLIBC_2.33 f32divf64x F
+GLIBC_2.33 f32mulf128 F
+GLIBC_2.33 f32mulf32x F
+GLIBC_2.33 f32mulf64 F
+GLIBC_2.33 f32mulf64x F
+GLIBC_2.33 f32subf128 F
+GLIBC_2.33 f32subf32x F
+GLIBC_2.33 f32subf64 F
+GLIBC_2.33 f32subf64x F
+GLIBC_2.33 f32xaddf128 F
+GLIBC_2.33 f32xaddf64 F
+GLIBC_2.33 f32xaddf64x F
+GLIBC_2.33 f32xdivf128 F
+GLIBC_2.33 f32xdivf64 F
+GLIBC_2.33 f32xdivf64x F
+GLIBC_2.33 f32xmulf128 F
+GLIBC_2.33 f32xmulf64 F
+GLIBC_2.33 f32xmulf64x F
+GLIBC_2.33 f32xsubf128 F
+GLIBC_2.33 f32xsubf64 F
+GLIBC_2.33 f32xsubf64x F
+GLIBC_2.33 f64addf128 F
+GLIBC_2.33 f64addf64x F
+GLIBC_2.33 f64divf128 F
+GLIBC_2.33 f64divf64x F
+GLIBC_2.33 f64mulf128 F
+GLIBC_2.33 f64mulf64x F
+GLIBC_2.33 f64subf128 F
+GLIBC_2.33 f64subf64x F
+GLIBC_2.33 f64xaddf128 F
+GLIBC_2.33 f64xdivf128 F
+GLIBC_2.33 f64xmulf128 F
+GLIBC_2.33 f64xsubf128 F
+GLIBC_2.33 fabs F
+GLIBC_2.33 fabsf F
+GLIBC_2.33 fabsf128 F
+GLIBC_2.33 fabsf32 F
+GLIBC_2.33 fabsf32x F
+GLIBC_2.33 fabsf64 F
+GLIBC_2.33 fabsf64x F
+GLIBC_2.33 fabsl F
+GLIBC_2.33 fadd F
+GLIBC_2.33 faddl F
+GLIBC_2.33 fdim F
+GLIBC_2.33 fdimf F
+GLIBC_2.33 fdimf128 F
+GLIBC_2.33 fdimf32 F
+GLIBC_2.33 fdimf32x F
+GLIBC_2.33 fdimf64 F
+GLIBC_2.33 fdimf64x F
+GLIBC_2.33 fdiml F
+GLIBC_2.33 fdiv F
+GLIBC_2.33 fdivl F
+GLIBC_2.33 feclearexcept F
+GLIBC_2.33 fedisableexcept F
+GLIBC_2.33 feenableexcept F
+GLIBC_2.33 fegetenv F
+GLIBC_2.33 fegetexcept F
+GLIBC_2.33 fegetexceptflag F
+GLIBC_2.33 fegetmode F
+GLIBC_2.33 fegetround F
+GLIBC_2.33 feholdexcept F
+GLIBC_2.33 feraiseexcept F
+GLIBC_2.33 fesetenv F
+GLIBC_2.33 fesetexcept F
+GLIBC_2.33 fesetexceptflag F
+GLIBC_2.33 fesetmode F
+GLIBC_2.33 fesetround F
+GLIBC_2.33 fetestexcept F
+GLIBC_2.33 fetestexceptflag F
+GLIBC_2.33 feupdateenv F
+GLIBC_2.33 finite F
+GLIBC_2.33 finitef F
+GLIBC_2.33 finitel F
+GLIBC_2.33 floor F
+GLIBC_2.33 floorf F
+GLIBC_2.33 floorf128 F
+GLIBC_2.33 floorf32 F
+GLIBC_2.33 floorf32x F
+GLIBC_2.33 floorf64 F
+GLIBC_2.33 floorf64x F
+GLIBC_2.33 floorl F
+GLIBC_2.33 fma F
+GLIBC_2.33 fmaf F
+GLIBC_2.33 fmaf128 F
+GLIBC_2.33 fmaf32 F
+GLIBC_2.33 fmaf32x F
+GLIBC_2.33 fmaf64 F
+GLIBC_2.33 fmaf64x F
+GLIBC_2.33 fmal F
+GLIBC_2.33 fmax F
+GLIBC_2.33 fmaxf F
+GLIBC_2.33 fmaxf128 F
+GLIBC_2.33 fmaxf32 F
+GLIBC_2.33 fmaxf32x F
+GLIBC_2.33 fmaxf64 F
+GLIBC_2.33 fmaxf64x F
+GLIBC_2.33 fmaxl F
+GLIBC_2.33 fmaxmag F
+GLIBC_2.33 fmaxmagf F
+GLIBC_2.33 fmaxmagf128 F
+GLIBC_2.33 fmaxmagf32 F
+GLIBC_2.33 fmaxmagf32x F
+GLIBC_2.33 fmaxmagf64 F
+GLIBC_2.33 fmaxmagf64x F
+GLIBC_2.33 fmaxmagl F
+GLIBC_2.33 fmin F
+GLIBC_2.33 fminf F
+GLIBC_2.33 fminf128 F
+GLIBC_2.33 fminf32 F
+GLIBC_2.33 fminf32x F
+GLIBC_2.33 fminf64 F
+GLIBC_2.33 fminf64x F
+GLIBC_2.33 fminl F
+GLIBC_2.33 fminmag F
+GLIBC_2.33 fminmagf F
+GLIBC_2.33 fminmagf128 F
+GLIBC_2.33 fminmagf32 F
+GLIBC_2.33 fminmagf32x F
+GLIBC_2.33 fminmagf64 F
+GLIBC_2.33 fminmagf64x F
+GLIBC_2.33 fminmagl F
+GLIBC_2.33 fmod F
+GLIBC_2.33 fmodf F
+GLIBC_2.33 fmodf128 F
+GLIBC_2.33 fmodf32 F
+GLIBC_2.33 fmodf32x F
+GLIBC_2.33 fmodf64 F
+GLIBC_2.33 fmodf64x F
+GLIBC_2.33 fmodl F
+GLIBC_2.33 fmul F
+GLIBC_2.33 fmull F
+GLIBC_2.33 frexp F
+GLIBC_2.33 frexpf F
+GLIBC_2.33 frexpf128 F
+GLIBC_2.33 frexpf32 F
+GLIBC_2.33 frexpf32x F
+GLIBC_2.33 frexpf64 F
+GLIBC_2.33 frexpf64x F
+GLIBC_2.33 frexpl F
+GLIBC_2.33 fromfp F
+GLIBC_2.33 fromfpf F
+GLIBC_2.33 fromfpf128 F
+GLIBC_2.33 fromfpf32 F
+GLIBC_2.33 fromfpf32x F
+GLIBC_2.33 fromfpf64 F
+GLIBC_2.33 fromfpf64x F
+GLIBC_2.33 fromfpl F
+GLIBC_2.33 fromfpx F
+GLIBC_2.33 fromfpxf F
+GLIBC_2.33 fromfpxf128 F
+GLIBC_2.33 fromfpxf32 F
+GLIBC_2.33 fromfpxf32x F
+GLIBC_2.33 fromfpxf64 F
+GLIBC_2.33 fromfpxf64x F
+GLIBC_2.33 fromfpxl F
+GLIBC_2.33 fsub F
+GLIBC_2.33 fsubl F
+GLIBC_2.33 gamma F
+GLIBC_2.33 gammaf F
+GLIBC_2.33 gammal F
+GLIBC_2.33 getpayload F
+GLIBC_2.33 getpayloadf F
+GLIBC_2.33 getpayloadf128 F
+GLIBC_2.33 getpayloadf32 F
+GLIBC_2.33 getpayloadf32x F
+GLIBC_2.33 getpayloadf64 F
+GLIBC_2.33 getpayloadf64x F
+GLIBC_2.33 getpayloadl F
+GLIBC_2.33 hypot F
+GLIBC_2.33 hypotf F
+GLIBC_2.33 hypotf128 F
+GLIBC_2.33 hypotf32 F
+GLIBC_2.33 hypotf32x F
+GLIBC_2.33 hypotf64 F
+GLIBC_2.33 hypotf64x F
+GLIBC_2.33 hypotl F
+GLIBC_2.33 ilogb F
+GLIBC_2.33 ilogbf F
+GLIBC_2.33 ilogbf128 F
+GLIBC_2.33 ilogbf32 F
+GLIBC_2.33 ilogbf32x F
+GLIBC_2.33 ilogbf64 F
+GLIBC_2.33 ilogbf64x F
+GLIBC_2.33 ilogbl F
+GLIBC_2.33 j0 F
+GLIBC_2.33 j0f F
+GLIBC_2.33 j0f128 F
+GLIBC_2.33 j0f32 F
+GLIBC_2.33 j0f32x F
+GLIBC_2.33 j0f64 F
+GLIBC_2.33 j0f64x F
+GLIBC_2.33 j0l F
+GLIBC_2.33 j1 F
+GLIBC_2.33 j1f F
+GLIBC_2.33 j1f128 F
+GLIBC_2.33 j1f32 F
+GLIBC_2.33 j1f32x F
+GLIBC_2.33 j1f64 F
+GLIBC_2.33 j1f64x F
+GLIBC_2.33 j1l F
+GLIBC_2.33 jn F
+GLIBC_2.33 jnf F
+GLIBC_2.33 jnf128 F
+GLIBC_2.33 jnf32 F
+GLIBC_2.33 jnf32x F
+GLIBC_2.33 jnf64 F
+GLIBC_2.33 jnf64x F
+GLIBC_2.33 jnl F
+GLIBC_2.33 ldexp F
+GLIBC_2.33 ldexpf F
+GLIBC_2.33 ldexpf128 F
+GLIBC_2.33 ldexpf32 F
+GLIBC_2.33 ldexpf32x F
+GLIBC_2.33 ldexpf64 F
+GLIBC_2.33 ldexpf64x F
+GLIBC_2.33 ldexpl F
+GLIBC_2.33 lgamma F
+GLIBC_2.33 lgamma_r F
+GLIBC_2.33 lgammaf F
+GLIBC_2.33 lgammaf128 F
+GLIBC_2.33 lgammaf128_r F
+GLIBC_2.33 lgammaf32 F
+GLIBC_2.33 lgammaf32_r F
+GLIBC_2.33 lgammaf32x F
+GLIBC_2.33 lgammaf32x_r F
+GLIBC_2.33 lgammaf64 F
+GLIBC_2.33 lgammaf64_r F
+GLIBC_2.33 lgammaf64x F
+GLIBC_2.33 lgammaf64x_r F
+GLIBC_2.33 lgammaf_r F
+GLIBC_2.33 lgammal F
+GLIBC_2.33 lgammal_r F
+GLIBC_2.33 llogb F
+GLIBC_2.33 llogbf F
+GLIBC_2.33 llogbf128 F
+GLIBC_2.33 llogbf32 F
+GLIBC_2.33 llogbf32x F
+GLIBC_2.33 llogbf64 F
+GLIBC_2.33 llogbf64x F
+GLIBC_2.33 llogbl F
+GLIBC_2.33 llrint F
+GLIBC_2.33 llrintf F
+GLIBC_2.33 llrintf128 F
+GLIBC_2.33 llrintf32 F
+GLIBC_2.33 llrintf32x F
+GLIBC_2.33 llrintf64 F
+GLIBC_2.33 llrintf64x F
+GLIBC_2.33 llrintl F
+GLIBC_2.33 llround F
+GLIBC_2.33 llroundf F
+GLIBC_2.33 llroundf128 F
+GLIBC_2.33 llroundf32 F
+GLIBC_2.33 llroundf32x F
+GLIBC_2.33 llroundf64 F
+GLIBC_2.33 llroundf64x F
+GLIBC_2.33 llroundl F
+GLIBC_2.33 log F
+GLIBC_2.33 log10 F
+GLIBC_2.33 log10f F
+GLIBC_2.33 log10f128 F
+GLIBC_2.33 log10f32 F
+GLIBC_2.33 log10f32x F
+GLIBC_2.33 log10f64 F
+GLIBC_2.33 log10f64x F
+GLIBC_2.33 log10l F
+GLIBC_2.33 log1p F
+GLIBC_2.33 log1pf F
+GLIBC_2.33 log1pf128 F
+GLIBC_2.33 log1pf32 F
+GLIBC_2.33 log1pf32x F
+GLIBC_2.33 log1pf64 F
+GLIBC_2.33 log1pf64x F
+GLIBC_2.33 log1pl F
+GLIBC_2.33 log2 F
+GLIBC_2.33 log2f F
+GLIBC_2.33 log2f128 F
+GLIBC_2.33 log2f32 F
+GLIBC_2.33 log2f32x F
+GLIBC_2.33 log2f64 F
+GLIBC_2.33 log2f64x F
+GLIBC_2.33 log2l F
+GLIBC_2.33 logb F
+GLIBC_2.33 logbf F
+GLIBC_2.33 logbf128 F
+GLIBC_2.33 logbf32 F
+GLIBC_2.33 logbf32x F
+GLIBC_2.33 logbf64 F
+GLIBC_2.33 logbf64x F
+GLIBC_2.33 logbl F
+GLIBC_2.33 logf F
+GLIBC_2.33 logf128 F
+GLIBC_2.33 logf32 F
+GLIBC_2.33 logf32x F
+GLIBC_2.33 logf64 F
+GLIBC_2.33 logf64x F
+GLIBC_2.33 logl F
+GLIBC_2.33 lrint F
+GLIBC_2.33 lrintf F
+GLIBC_2.33 lrintf128 F
+GLIBC_2.33 lrintf32 F
+GLIBC_2.33 lrintf32x F
+GLIBC_2.33 lrintf64 F
+GLIBC_2.33 lrintf64x F
+GLIBC_2.33 lrintl F
+GLIBC_2.33 lround F
+GLIBC_2.33 lroundf F
+GLIBC_2.33 lroundf128 F
+GLIBC_2.33 lroundf32 F
+GLIBC_2.33 lroundf32x F
+GLIBC_2.33 lroundf64 F
+GLIBC_2.33 lroundf64x F
+GLIBC_2.33 lroundl F
+GLIBC_2.33 modf F
+GLIBC_2.33 modff F
+GLIBC_2.33 modff128 F
+GLIBC_2.33 modff32 F
+GLIBC_2.33 modff32x F
+GLIBC_2.33 modff64 F
+GLIBC_2.33 modff64x F
+GLIBC_2.33 modfl F
+GLIBC_2.33 nan F
+GLIBC_2.33 nanf F
+GLIBC_2.33 nanf128 F
+GLIBC_2.33 nanf32 F
+GLIBC_2.33 nanf32x F
+GLIBC_2.33 nanf64 F
+GLIBC_2.33 nanf64x F
+GLIBC_2.33 nanl F
+GLIBC_2.33 nearbyint F
+GLIBC_2.33 nearbyintf F
+GLIBC_2.33 nearbyintf128 F
+GLIBC_2.33 nearbyintf32 F
+GLIBC_2.33 nearbyintf32x F
+GLIBC_2.33 nearbyintf64 F
+GLIBC_2.33 nearbyintf64x F
+GLIBC_2.33 nearbyintl F
+GLIBC_2.33 nextafter F
+GLIBC_2.33 nextafterf F
+GLIBC_2.33 nextafterf128 F
+GLIBC_2.33 nextafterf32 F
+GLIBC_2.33 nextafterf32x F
+GLIBC_2.33 nextafterf64 F
+GLIBC_2.33 nextafterf64x F
+GLIBC_2.33 nextafterl F
+GLIBC_2.33 nextdown F
+GLIBC_2.33 nextdownf F
+GLIBC_2.33 nextdownf128 F
+GLIBC_2.33 nextdownf32 F
+GLIBC_2.33 nextdownf32x F
+GLIBC_2.33 nextdownf64 F
+GLIBC_2.33 nextdownf64x F
+GLIBC_2.33 nextdownl F
+GLIBC_2.33 nexttoward F
+GLIBC_2.33 nexttowardf F
+GLIBC_2.33 nexttowardl F
+GLIBC_2.33 nextup F
+GLIBC_2.33 nextupf F
+GLIBC_2.33 nextupf128 F
+GLIBC_2.33 nextupf32 F
+GLIBC_2.33 nextupf32x F
+GLIBC_2.33 nextupf64 F
+GLIBC_2.33 nextupf64x F
+GLIBC_2.33 nextupl F
+GLIBC_2.33 pow F
+GLIBC_2.33 powf F
+GLIBC_2.33 powf128 F
+GLIBC_2.33 powf32 F
+GLIBC_2.33 powf32x F
+GLIBC_2.33 powf64 F
+GLIBC_2.33 powf64x F
+GLIBC_2.33 powl F
+GLIBC_2.33 remainder F
+GLIBC_2.33 remainderf F
+GLIBC_2.33 remainderf128 F
+GLIBC_2.33 remainderf32 F
+GLIBC_2.33 remainderf32x F
+GLIBC_2.33 remainderf64 F
+GLIBC_2.33 remainderf64x F
+GLIBC_2.33 remainderl F
+GLIBC_2.33 remquo F
+GLIBC_2.33 remquof F
+GLIBC_2.33 remquof128 F
+GLIBC_2.33 remquof32 F
+GLIBC_2.33 remquof32x F
+GLIBC_2.33 remquof64 F
+GLIBC_2.33 remquof64x F
+GLIBC_2.33 remquol F
+GLIBC_2.33 rint F
+GLIBC_2.33 rintf F
+GLIBC_2.33 rintf128 F
+GLIBC_2.33 rintf32 F
+GLIBC_2.33 rintf32x F
+GLIBC_2.33 rintf64 F
+GLIBC_2.33 rintf64x F
+GLIBC_2.33 rintl F
+GLIBC_2.33 round F
+GLIBC_2.33 roundeven F
+GLIBC_2.33 roundevenf F
+GLIBC_2.33 roundevenf128 F
+GLIBC_2.33 roundevenf32 F
+GLIBC_2.33 roundevenf32x F
+GLIBC_2.33 roundevenf64 F
+GLIBC_2.33 roundevenf64x F
+GLIBC_2.33 roundevenl F
+GLIBC_2.33 roundf F
+GLIBC_2.33 roundf128 F
+GLIBC_2.33 roundf32 F
+GLIBC_2.33 roundf32x F
+GLIBC_2.33 roundf64 F
+GLIBC_2.33 roundf64x F
+GLIBC_2.33 roundl F
+GLIBC_2.33 scalb F
+GLIBC_2.33 scalbf F
+GLIBC_2.33 scalbl F
+GLIBC_2.33 scalbln F
+GLIBC_2.33 scalblnf F
+GLIBC_2.33 scalblnf128 F
+GLIBC_2.33 scalblnf32 F
+GLIBC_2.33 scalblnf32x F
+GLIBC_2.33 scalblnf64 F
+GLIBC_2.33 scalblnf64x F
+GLIBC_2.33 scalblnl F
+GLIBC_2.33 scalbn F
+GLIBC_2.33 scalbnf F
+GLIBC_2.33 scalbnf128 F
+GLIBC_2.33 scalbnf32 F
+GLIBC_2.33 scalbnf32x F
+GLIBC_2.33 scalbnf64 F
+GLIBC_2.33 scalbnf64x F
+GLIBC_2.33 scalbnl F
+GLIBC_2.33 setpayload F
+GLIBC_2.33 setpayloadf F
+GLIBC_2.33 setpayloadf128 F
+GLIBC_2.33 setpayloadf32 F
+GLIBC_2.33 setpayloadf32x F
+GLIBC_2.33 setpayloadf64 F
+GLIBC_2.33 setpayloadf64x F
+GLIBC_2.33 setpayloadl F
+GLIBC_2.33 setpayloadsig F
+GLIBC_2.33 setpayloadsigf F
+GLIBC_2.33 setpayloadsigf128 F
+GLIBC_2.33 setpayloadsigf32 F
+GLIBC_2.33 setpayloadsigf32x F
+GLIBC_2.33 setpayloadsigf64 F
+GLIBC_2.33 setpayloadsigf64x F
+GLIBC_2.33 setpayloadsigl F
+GLIBC_2.33 signgam D 0x4
+GLIBC_2.33 significand F
+GLIBC_2.33 significandf F
+GLIBC_2.33 significandl F
+GLIBC_2.33 sin F
+GLIBC_2.33 sincos F
+GLIBC_2.33 sincosf F
+GLIBC_2.33 sincosf128 F
+GLIBC_2.33 sincosf32 F
+GLIBC_2.33 sincosf32x F
+GLIBC_2.33 sincosf64 F
+GLIBC_2.33 sincosf64x F
+GLIBC_2.33 sincosl F
+GLIBC_2.33 sinf F
+GLIBC_2.33 sinf128 F
+GLIBC_2.33 sinf32 F
+GLIBC_2.33 sinf32x F
+GLIBC_2.33 sinf64 F
+GLIBC_2.33 sinf64x F
+GLIBC_2.33 sinh F
+GLIBC_2.33 sinhf F
+GLIBC_2.33 sinhf128 F
+GLIBC_2.33 sinhf32 F
+GLIBC_2.33 sinhf32x F
+GLIBC_2.33 sinhf64 F
+GLIBC_2.33 sinhf64x F
+GLIBC_2.33 sinhl F
+GLIBC_2.33 sinl F
+GLIBC_2.33 sqrt F
+GLIBC_2.33 sqrtf F
+GLIBC_2.33 sqrtf128 F
+GLIBC_2.33 sqrtf32 F
+GLIBC_2.33 sqrtf32x F
+GLIBC_2.33 sqrtf64 F
+GLIBC_2.33 sqrtf64x F
+GLIBC_2.33 sqrtl F
+GLIBC_2.33 tan F
+GLIBC_2.33 tanf F
+GLIBC_2.33 tanf128 F
+GLIBC_2.33 tanf32 F
+GLIBC_2.33 tanf32x F
+GLIBC_2.33 tanf64 F
+GLIBC_2.33 tanf64x F
+GLIBC_2.33 tanh F
+GLIBC_2.33 tanhf F
+GLIBC_2.33 tanhf128 F
+GLIBC_2.33 tanhf32 F
+GLIBC_2.33 tanhf32x F
+GLIBC_2.33 tanhf64 F
+GLIBC_2.33 tanhf64x F
+GLIBC_2.33 tanhl F
+GLIBC_2.33 tanl F
+GLIBC_2.33 tgamma F
+GLIBC_2.33 tgammaf F
+GLIBC_2.33 tgammaf128 F
+GLIBC_2.33 tgammaf32 F
+GLIBC_2.33 tgammaf32x F
+GLIBC_2.33 tgammaf64 F
+GLIBC_2.33 tgammaf64x F
+GLIBC_2.33 tgammal F
+GLIBC_2.33 totalorder F
+GLIBC_2.33 totalorderf F
+GLIBC_2.33 totalorderf128 F
+GLIBC_2.33 totalorderf32 F
+GLIBC_2.33 totalorderf32x F
+GLIBC_2.33 totalorderf64 F
+GLIBC_2.33 totalorderf64x F
+GLIBC_2.33 totalorderl F
+GLIBC_2.33 totalordermag F
+GLIBC_2.33 totalordermagf F
+GLIBC_2.33 totalordermagf128 F
+GLIBC_2.33 totalordermagf32 F
+GLIBC_2.33 totalordermagf32x F
+GLIBC_2.33 totalordermagf64 F
+GLIBC_2.33 totalordermagf64x F
+GLIBC_2.33 totalordermagl F
+GLIBC_2.33 trunc F
+GLIBC_2.33 truncf F
+GLIBC_2.33 truncf128 F
+GLIBC_2.33 truncf32 F
+GLIBC_2.33 truncf32x F
+GLIBC_2.33 truncf64 F
+GLIBC_2.33 truncf64x F
+GLIBC_2.33 truncl F
+GLIBC_2.33 ufromfp F
+GLIBC_2.33 ufromfpf F
+GLIBC_2.33 ufromfpf128 F
+GLIBC_2.33 ufromfpf32 F
+GLIBC_2.33 ufromfpf32x F
+GLIBC_2.33 ufromfpf64 F
+GLIBC_2.33 ufromfpf64x F
+GLIBC_2.33 ufromfpl F
+GLIBC_2.33 ufromfpx F
+GLIBC_2.33 ufromfpxf F
+GLIBC_2.33 ufromfpxf128 F
+GLIBC_2.33 ufromfpxf32 F
+GLIBC_2.33 ufromfpxf32x F
+GLIBC_2.33 ufromfpxf64 F
+GLIBC_2.33 ufromfpxf64x F
+GLIBC_2.33 ufromfpxl F
+GLIBC_2.33 y0 F
+GLIBC_2.33 y0f F
+GLIBC_2.33 y0f128 F
+GLIBC_2.33 y0f32 F
+GLIBC_2.33 y0f32x F
+GLIBC_2.33 y0f64 F
+GLIBC_2.33 y0f64x F
+GLIBC_2.33 y0l F
+GLIBC_2.33 y1 F
+GLIBC_2.33 y1f F
+GLIBC_2.33 y1f128 F
+GLIBC_2.33 y1f32 F
+GLIBC_2.33 y1f32x F
+GLIBC_2.33 y1f64 F
+GLIBC_2.33 y1f64x F
+GLIBC_2.33 y1l F
+GLIBC_2.33 yn F
+GLIBC_2.33 ynf F
+GLIBC_2.33 ynf128 F
+GLIBC_2.33 ynf32 F
+GLIBC_2.33 ynf32x F
+GLIBC_2.33 ynf64 F
+GLIBC_2.33 ynf64x F
+GLIBC_2.33 ynl F
diff --git a/sysdeps/unix/sysv/linux/riscv/rv32/libpthread.abilist b/sysdeps/unix/sysv/linux/riscv/rv32/libpthread.abilist
new file mode 100644
index 0000000000..61b3c4ff7a
--- /dev/null
+++ b/sysdeps/unix/sysv/linux/riscv/rv32/libpthread.abilist
@@ -0,0 +1,213 @@
+GLIBC_2.33 _IO_flockfile F
+GLIBC_2.33 _IO_ftrylockfile F
+GLIBC_2.33 _IO_funlockfile F
+GLIBC_2.33 __close F
+GLIBC_2.33 __connect F
+GLIBC_2.33 __errno_location F
+GLIBC_2.33 __h_errno_location F
+GLIBC_2.33 __libc_allocate_rtsig F
+GLIBC_2.33 __libc_current_sigrtmax F
+GLIBC_2.33 __libc_current_sigrtmin F
+GLIBC_2.33 __lseek F
+GLIBC_2.33 __open F
+GLIBC_2.33 __open64 F
+GLIBC_2.33 __pread64 F
+GLIBC_2.33 __pthread_cleanup_routine F
+GLIBC_2.33 __pthread_getspecific F
+GLIBC_2.33 __pthread_key_create F
+GLIBC_2.33 __pthread_mutex_destroy F
+GLIBC_2.33 __pthread_mutex_init F
+GLIBC_2.33 __pthread_mutex_lock F
+GLIBC_2.33 __pthread_mutex_trylock F
+GLIBC_2.33 __pthread_mutex_unlock F
+GLIBC_2.33 __pthread_mutexattr_destroy F
+GLIBC_2.33 __pthread_mutexattr_init F
+GLIBC_2.33 __pthread_mutexattr_settype F
+GLIBC_2.33 __pthread_once F
+GLIBC_2.33 __pthread_register_cancel F
+GLIBC_2.33 __pthread_register_cancel_defer F
+GLIBC_2.33 __pthread_rwlock_destroy F
+GLIBC_2.33 __pthread_rwlock_init F
+GLIBC_2.33 __pthread_rwlock_rdlock F
+GLIBC_2.33 __pthread_rwlock_tryrdlock F
+GLIBC_2.33 __pthread_rwlock_trywrlock F
+GLIBC_2.33 __pthread_rwlock_unlock F
+GLIBC_2.33 __pthread_rwlock_wrlock F
+GLIBC_2.33 __pthread_setspecific F
+GLIBC_2.33 __pthread_unregister_cancel F
+GLIBC_2.33 __pthread_unregister_cancel_restore F
+GLIBC_2.33 __pthread_unwind_next F
+GLIBC_2.33 __pwrite64 F
+GLIBC_2.33 __read F
+GLIBC_2.33 __res_state F
+GLIBC_2.33 __send F
+GLIBC_2.33 __sigaction F
+GLIBC_2.33 __write F
+GLIBC_2.33 _pthread_cleanup_pop F
+GLIBC_2.33 _pthread_cleanup_pop_restore F
+GLIBC_2.33 _pthread_cleanup_push F
+GLIBC_2.33 _pthread_cleanup_push_defer F
+GLIBC_2.33 accept F
+GLIBC_2.33 call_once F
+GLIBC_2.33 close F
+GLIBC_2.33 cnd_broadcast F
+GLIBC_2.33 cnd_destroy F
+GLIBC_2.33 cnd_init F
+GLIBC_2.33 cnd_signal F
+GLIBC_2.33 cnd_timedwait F
+GLIBC_2.33 cnd_wait F
+GLIBC_2.33 connect F
+GLIBC_2.33 flockfile F
+GLIBC_2.33 fsync F
+GLIBC_2.33 ftrylockfile F
+GLIBC_2.33 funlockfile F
+GLIBC_2.33 lseek F
+GLIBC_2.33 lseek64 F
+GLIBC_2.33 msync F
+GLIBC_2.33 mtx_destroy F
+GLIBC_2.33 mtx_init F
+GLIBC_2.33 mtx_lock F
+GLIBC_2.33 mtx_timedlock F
+GLIBC_2.33 mtx_trylock F
+GLIBC_2.33 mtx_unlock F
+GLIBC_2.33 open F
+GLIBC_2.33 open64 F
+GLIBC_2.33 pause F
+GLIBC_2.33 pread F
+GLIBC_2.33 pread64 F
+GLIBC_2.33 pthread_attr_getaffinity_np F
+GLIBC_2.33 pthread_attr_getguardsize F
+GLIBC_2.33 pthread_attr_getstack F
+GLIBC_2.33 pthread_attr_getstackaddr F
+GLIBC_2.33 pthread_attr_getstacksize F
+GLIBC_2.33 pthread_attr_setguardsize F
+GLIBC_2.33 pthread_attr_setstack F
+GLIBC_2.33 pthread_attr_setstackaddr F
+GLIBC_2.33 pthread_attr_setstacksize F
+GLIBC_2.33 pthread_barrier_destroy F
+GLIBC_2.33 pthread_barrier_init F
+GLIBC_2.33 pthread_barrier_wait F
+GLIBC_2.33 pthread_barrierattr_destroy F
+GLIBC_2.33 pthread_barrierattr_getpshared F
+GLIBC_2.33 pthread_barrierattr_init F
+GLIBC_2.33 pthread_barrierattr_setpshared F
+GLIBC_2.33 pthread_cancel F
+GLIBC_2.33 pthread_clockjoin_np F
+GLIBC_2.33 pthread_cond_broadcast F
+GLIBC_2.33 pthread_cond_clockwait F
+GLIBC_2.33 pthread_cond_signal F
+GLIBC_2.33 pthread_cond_timedwait F
+GLIBC_2.33 pthread_cond_wait F
+GLIBC_2.33 pthread_condattr_getclock F
+GLIBC_2.33 pthread_condattr_getpshared F
+GLIBC_2.33 pthread_condattr_setclock F
+GLIBC_2.33 pthread_condattr_setpshared F
+GLIBC_2.33 pthread_create F
+GLIBC_2.33 pthread_detach F
+GLIBC_2.33 pthread_exit F
+GLIBC_2.33 pthread_getattr_default_np F
+GLIBC_2.33 pthread_getconcurrency F
+GLIBC_2.33 pthread_getcpuclockid F
+GLIBC_2.33 pthread_getname_np F
+GLIBC_2.33 pthread_getspecific F
+GLIBC_2.33 pthread_join F
+GLIBC_2.33 pthread_key_create F
+GLIBC_2.33 pthread_key_delete F
+GLIBC_2.33 pthread_kill F
+GLIBC_2.33 pthread_kill_other_threads_np F
+GLIBC_2.33 pthread_mutex_clocklock F
+GLIBC_2.33 pthread_mutex_consistent F
+GLIBC_2.33 pthread_mutex_consistent_np F
+GLIBC_2.33 pthread_mutex_destroy F
+GLIBC_2.33 pthread_mutex_getprioceiling F
+GLIBC_2.33 pthread_mutex_init F
+GLIBC_2.33 pthread_mutex_lock F
+GLIBC_2.33 pthread_mutex_setprioceiling F
+GLIBC_2.33 pthread_mutex_timedlock F
+GLIBC_2.33 pthread_mutex_trylock F
+GLIBC_2.33 pthread_mutex_unlock F
+GLIBC_2.33 pthread_mutexattr_destroy F
+GLIBC_2.33 pthread_mutexattr_getkind_np F
+GLIBC_2.33 pthread_mutexattr_getprioceiling F
+GLIBC_2.33 pthread_mutexattr_getprotocol F
+GLIBC_2.33 pthread_mutexattr_getpshared F
+GLIBC_2.33 pthread_mutexattr_getrobust F
+GLIBC_2.33 pthread_mutexattr_getrobust_np F
+GLIBC_2.33 pthread_mutexattr_gettype F
+GLIBC_2.33 pthread_mutexattr_init F
+GLIBC_2.33 pthread_mutexattr_setkind_np F
+GLIBC_2.33 pthread_mutexattr_setprioceiling F
+GLIBC_2.33 pthread_mutexattr_setprotocol F
+GLIBC_2.33 pthread_mutexattr_setpshared F
+GLIBC_2.33 pthread_mutexattr_setrobust F
+GLIBC_2.33 pthread_mutexattr_setrobust_np F
+GLIBC_2.33 pthread_mutexattr_settype F
+GLIBC_2.33 pthread_once F
+GLIBC_2.33 pthread_rwlock_clockrdlock F
+GLIBC_2.33 pthread_rwlock_clockwrlock F
+GLIBC_2.33 pthread_rwlock_destroy F
+GLIBC_2.33 pthread_rwlock_init F
+GLIBC_2.33 pthread_rwlock_rdlock F
+GLIBC_2.33 pthread_rwlock_timedrdlock F
+GLIBC_2.33 pthread_rwlock_timedwrlock F
+GLIBC_2.33 pthread_rwlock_tryrdlock F
+GLIBC_2.33 pthread_rwlock_trywrlock F
+GLIBC_2.33 pthread_rwlock_unlock F
+GLIBC_2.33 pthread_rwlock_wrlock F
+GLIBC_2.33 pthread_rwlockattr_destroy F
+GLIBC_2.33 pthread_rwlockattr_getkind_np F
+GLIBC_2.33 pthread_rwlockattr_getpshared F
+GLIBC_2.33 pthread_rwlockattr_init F
+GLIBC_2.33 pthread_rwlockattr_setkind_np F
+GLIBC_2.33 pthread_rwlockattr_setpshared F
+GLIBC_2.33 pthread_setaffinity_np F
+GLIBC_2.33 pthread_setattr_default_np F
+GLIBC_2.33 pthread_setcancelstate F
+GLIBC_2.33 pthread_setcanceltype F
+GLIBC_2.33 pthread_setconcurrency F
+GLIBC_2.33 pthread_setname_np F
+GLIBC_2.33 pthread_setschedprio F
+GLIBC_2.33 pthread_setspecific F
+GLIBC_2.33 pthread_sigqueue F
+GLIBC_2.33 pthread_spin_destroy F
+GLIBC_2.33 pthread_spin_init F
+GLIBC_2.33 pthread_spin_lock F
+GLIBC_2.33 pthread_spin_trylock F
+GLIBC_2.33 pthread_spin_unlock F
+GLIBC_2.33 pthread_testcancel F
+GLIBC_2.33 pthread_timedjoin_np F
+GLIBC_2.33 pthread_tryjoin_np F
+GLIBC_2.33 pthread_yield F
+GLIBC_2.33 pwrite F
+GLIBC_2.33 pwrite64 F
+GLIBC_2.33 raise F
+GLIBC_2.33 read F
+GLIBC_2.33 recv F
+GLIBC_2.33 recvfrom F
+GLIBC_2.33 recvmsg F
+GLIBC_2.33 sem_clockwait F
+GLIBC_2.33 sem_close F
+GLIBC_2.33 sem_destroy F
+GLIBC_2.33 sem_getvalue F
+GLIBC_2.33 sem_init F
+GLIBC_2.33 sem_open F
+GLIBC_2.33 sem_post F
+GLIBC_2.33 sem_timedwait F
+GLIBC_2.33 sem_trywait F
+GLIBC_2.33 sem_unlink F
+GLIBC_2.33 sem_wait F
+GLIBC_2.33 send F
+GLIBC_2.33 sendmsg F
+GLIBC_2.33 sendto F
+GLIBC_2.33 sigaction F
+GLIBC_2.33 sigwait F
+GLIBC_2.33 tcdrain F
+GLIBC_2.33 thrd_create F
+GLIBC_2.33 thrd_detach F
+GLIBC_2.33 thrd_exit F
+GLIBC_2.33 thrd_join F
+GLIBC_2.33 tss_create F
+GLIBC_2.33 tss_delete F
+GLIBC_2.33 tss_get F
+GLIBC_2.33 tss_set F
+GLIBC_2.33 write F
diff --git a/sysdeps/unix/sysv/linux/riscv/rv32/libresolv.abilist b/sysdeps/unix/sysv/linux/riscv/rv32/libresolv.abilist
new file mode 100644
index 0000000000..43947f7e27
--- /dev/null
+++ b/sysdeps/unix/sysv/linux/riscv/rv32/libresolv.abilist
@@ -0,0 +1,79 @@
+GLIBC_2.33 __b64_ntop F
+GLIBC_2.33 __b64_pton F
+GLIBC_2.33 __dn_comp F
+GLIBC_2.33 __dn_count_labels F
+GLIBC_2.33 __dn_expand F
+GLIBC_2.33 __dn_skipname F
+GLIBC_2.33 __fp_nquery F
+GLIBC_2.33 __fp_query F
+GLIBC_2.33 __fp_resstat F
+GLIBC_2.33 __hostalias F
+GLIBC_2.33 __loc_aton F
+GLIBC_2.33 __loc_ntoa F
+GLIBC_2.33 __p_cdname F
+GLIBC_2.33 __p_cdnname F
+GLIBC_2.33 __p_class F
+GLIBC_2.33 __p_class_syms D 0x54
+GLIBC_2.33 __p_fqname F
+GLIBC_2.33 __p_fqnname F
+GLIBC_2.33 __p_option F
+GLIBC_2.33 __p_query F
+GLIBC_2.33 __p_rcode F
+GLIBC_2.33 __p_time F
+GLIBC_2.33 __p_type F
+GLIBC_2.33 __p_type_syms D 0x228
+GLIBC_2.33 __putlong F
+GLIBC_2.33 __putshort F
+GLIBC_2.33 __res_close F
+GLIBC_2.33 __res_dnok F
+GLIBC_2.33 __res_hnok F
+GLIBC_2.33 __res_hostalias F
+GLIBC_2.33 __res_isourserver F
+GLIBC_2.33 __res_mailok F
+GLIBC_2.33 __res_mkquery F
+GLIBC_2.33 __res_nameinquery F
+GLIBC_2.33 __res_nmkquery F
+GLIBC_2.33 __res_nquery F
+GLIBC_2.33 __res_nquerydomain F
+GLIBC_2.33 __res_nsearch F
+GLIBC_2.33 __res_nsend F
+GLIBC_2.33 __res_ownok F
+GLIBC_2.33 __res_queriesmatch F
+GLIBC_2.33 __res_query F
+GLIBC_2.33 __res_querydomain F
+GLIBC_2.33 __res_search F
+GLIBC_2.33 __res_send F
+GLIBC_2.33 __sym_ntop F
+GLIBC_2.33 __sym_ntos F
+GLIBC_2.33 __sym_ston F
+GLIBC_2.33 _getlong F
+GLIBC_2.33 _getshort F
+GLIBC_2.33 inet_net_ntop F
+GLIBC_2.33 inet_net_pton F
+GLIBC_2.33 inet_neta F
+GLIBC_2.33 ns_datetosecs F
+GLIBC_2.33 ns_format_ttl F
+GLIBC_2.33 ns_get16 F
+GLIBC_2.33 ns_get32 F
+GLIBC_2.33 ns_initparse F
+GLIBC_2.33 ns_makecanon F
+GLIBC_2.33 ns_msg_getflag F
+GLIBC_2.33 ns_name_compress F
+GLIBC_2.33 ns_name_ntol F
+GLIBC_2.33 ns_name_ntop F
+GLIBC_2.33 ns_name_pack F
+GLIBC_2.33 ns_name_pton F
+GLIBC_2.33 ns_name_rollback F
+GLIBC_2.33 ns_name_skip F
+GLIBC_2.33 ns_name_uncompress F
+GLIBC_2.33 ns_name_unpack F
+GLIBC_2.33 ns_parse_ttl F
+GLIBC_2.33 ns_parserr F
+GLIBC_2.33 ns_put16 F
+GLIBC_2.33 ns_put32 F
+GLIBC_2.33 ns_samedomain F
+GLIBC_2.33 ns_samename F
+GLIBC_2.33 ns_skiprr F
+GLIBC_2.33 ns_sprintrr F
+GLIBC_2.33 ns_sprintrrf F
+GLIBC_2.33 ns_subdomain F
diff --git a/sysdeps/unix/sysv/linux/riscv/rv32/librt.abilist b/sysdeps/unix/sysv/linux/riscv/rv32/librt.abilist
new file mode 100644
index 0000000000..8a3f5ed516
--- /dev/null
+++ b/sysdeps/unix/sysv/linux/riscv/rv32/librt.abilist
@@ -0,0 +1,35 @@
+GLIBC_2.33 __mq_open_2 F
+GLIBC_2.33 aio_cancel F
+GLIBC_2.33 aio_cancel64 F
+GLIBC_2.33 aio_error F
+GLIBC_2.33 aio_error64 F
+GLIBC_2.33 aio_fsync F
+GLIBC_2.33 aio_fsync64 F
+GLIBC_2.33 aio_init F
+GLIBC_2.33 aio_read F
+GLIBC_2.33 aio_read64 F
+GLIBC_2.33 aio_return F
+GLIBC_2.33 aio_return64 F
+GLIBC_2.33 aio_suspend F
+GLIBC_2.33 aio_suspend64 F
+GLIBC_2.33 aio_write F
+GLIBC_2.33 aio_write64 F
+GLIBC_2.33 lio_listio F
+GLIBC_2.33 lio_listio64 F
+GLIBC_2.33 mq_close F
+GLIBC_2.33 mq_getattr F
+GLIBC_2.33 mq_notify F
+GLIBC_2.33 mq_open F
+GLIBC_2.33 mq_receive F
+GLIBC_2.33 mq_send F
+GLIBC_2.33 mq_setattr F
+GLIBC_2.33 mq_timedreceive F
+GLIBC_2.33 mq_timedsend F
+GLIBC_2.33 mq_unlink F
+GLIBC_2.33 shm_open F
+GLIBC_2.33 shm_unlink F
+GLIBC_2.33 timer_create F
+GLIBC_2.33 timer_delete F
+GLIBC_2.33 timer_getoverrun F
+GLIBC_2.33 timer_gettime F
+GLIBC_2.33 timer_settime F
diff --git a/sysdeps/unix/sysv/linux/riscv/rv32/libthread_db.abilist b/sysdeps/unix/sysv/linux/riscv/rv32/libthread_db.abilist
new file mode 100644
index 0000000000..50dde8f69b
--- /dev/null
+++ b/sysdeps/unix/sysv/linux/riscv/rv32/libthread_db.abilist
@@ -0,0 +1,40 @@
+GLIBC_2.33 td_init F
+GLIBC_2.33 td_log F
+GLIBC_2.33 td_symbol_list F
+GLIBC_2.33 td_ta_clear_event F
+GLIBC_2.33 td_ta_delete F
+GLIBC_2.33 td_ta_enable_stats F
+GLIBC_2.33 td_ta_event_addr F
+GLIBC_2.33 td_ta_event_getmsg F
+GLIBC_2.33 td_ta_get_nthreads F
+GLIBC_2.33 td_ta_get_ph F
+GLIBC_2.33 td_ta_get_stats F
+GLIBC_2.33 td_ta_map_id2thr F
+GLIBC_2.33 td_ta_map_lwp2thr F
+GLIBC_2.33 td_ta_new F
+GLIBC_2.33 td_ta_reset_stats F
+GLIBC_2.33 td_ta_set_event F
+GLIBC_2.33 td_ta_setconcurrency F
+GLIBC_2.33 td_ta_thr_iter F
+GLIBC_2.33 td_ta_tsd_iter F
+GLIBC_2.33 td_thr_clear_event F
+GLIBC_2.33 td_thr_dbresume F
+GLIBC_2.33 td_thr_dbsuspend F
+GLIBC_2.33 td_thr_event_enable F
+GLIBC_2.33 td_thr_event_getmsg F
+GLIBC_2.33 td_thr_get_info F
+GLIBC_2.33 td_thr_getfpregs F
+GLIBC_2.33 td_thr_getgregs F
+GLIBC_2.33 td_thr_getxregs F
+GLIBC_2.33 td_thr_getxregsize F
+GLIBC_2.33 td_thr_set_event F
+GLIBC_2.33 td_thr_setfpregs F
+GLIBC_2.33 td_thr_setgregs F
+GLIBC_2.33 td_thr_setprio F
+GLIBC_2.33 td_thr_setsigpending F
+GLIBC_2.33 td_thr_setxregs F
+GLIBC_2.33 td_thr_sigsetmask F
+GLIBC_2.33 td_thr_tls_get_addr F
+GLIBC_2.33 td_thr_tlsbase F
+GLIBC_2.33 td_thr_tsd F
+GLIBC_2.33 td_thr_validate F
diff --git a/sysdeps/unix/sysv/linux/riscv/rv32/libutil.abilist b/sysdeps/unix/sysv/linux/riscv/rv32/libutil.abilist
new file mode 100644
index 0000000000..59ae944bda
--- /dev/null
+++ b/sysdeps/unix/sysv/linux/riscv/rv32/libutil.abilist
@@ -0,0 +1,6 @@
+GLIBC_2.33 forkpty F
+GLIBC_2.33 login F
+GLIBC_2.33 login_tty F
+GLIBC_2.33 logout F
+GLIBC_2.33 logwtmp F
+GLIBC_2.33 openpty F
-- 
2.29.0

